{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "78ac94e6-bc0b-4462-8237-4c95ef2006bd",
   "metadata": {},
   "source": [
    "What I need to do here to make sure that there is no confusion about the positions of opposite homozygotes is to call SNPs using the joint calling with the 2K matrix."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7774fd01-b570-4167-a858-e92e4054f425",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ChrisC1\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC2\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC3\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC4\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC5\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC6\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/GATK-ChrisC6.sbatch\"\n",
      "[1] \"ChrisC7\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC8\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/GATK-ChrisC8.sbatch\"\n",
      "[1] \"YJNRC2\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC17\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC18\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"Batching up...\"\n",
      "[1] \"2 /home/jnrunge/data/trd/mapped_reads/scripts/GATK.list\"\n",
      "[1] \"[1] \\\"Currently running 0 jobs.\\\"\"               \n",
      "[2] \"[1] \\\"2 sbatches remaining! Running 11 more...\\\"\"\n",
      "[3] \"[1] \\\"Submitted batch job 483880\\\"\"              \n",
      "[4] \"[1] \\\"Submitted batch job 483881\\\"\"              \n"
     ]
    }
   ],
   "source": [
    "source(\"../../BrusselSprouts/scripts/functions.R\")\n",
    "\n",
    "\n",
    "scripts_dir=\"/home/jnrunge/data/trd/mapped_reads/scripts/\"\n",
    "\n",
    "initial_timedate=Sys.time()\n",
    "\n",
    "# what needs to be done is individually, and parallely genotype all crosses, then merge them into a DB, and once call the joint genotyping (once per batch of sequencing, so twice in total)\n",
    "\n",
    "bam_files=list.files(path=\"~/data/trd/mapped_reads\",pattern=\"JNRC[0-9]*.bam$\",full.names = TRUE)\n",
    "bam_files=c(bam_files, list.files(path=\"~/data/trd/mapped_reads\",pattern=\"ChrisC[0-9].bam$\",full.names = TRUE))\n",
    "bam_files=naturalsort(bam_files)\n",
    "crosses=unlist(lapply(basename(bam_files), getFirst_v2, split=\".\"))\n",
    "\n",
    "for(cross in crosses){\n",
    "    print(cross)\n",
    "\n",
    "    if(file.exists(paste0(\"/home/jnrunge/data/trd/mapped_reads/\",cross,\".g.vcf.gz\"))){\n",
    "        print(\"GVCF exists\")\n",
    "        next\n",
    "    }\n",
    "    file.create(running_file<-paste0(\"~/data/TRD/\",cross,\".runningGATK\"))\n",
    "    cmds=paste0(\"cd /home/jnrunge/data/trd/mapped_reads/ && gatk AddOrReplaceReadGroups -I \",cross,\".bam -O \",cross,\".ReadGroups.bam --RGID \",cross,\" --RGLB \",cross,\" --RGPL ILLUMINA --RGPU \",cross,\" --RGSM \",cross,\"\")\n",
    "    cmds=c(cmds,paste0(\"mv -f \",cross,\".ReadGroups.bam \",cross,\".bam\"))\n",
    "    cmds=c(cmds,paste0(\"samtools index \",cross,\".bam\"))\n",
    "    cmds=c(cmds,paste0(\"gatk HaplotypeCaller -R ../../TRD/R64_nucl.fasta -I \",cross,\".bam -O \",cross,\".g.vcf.gz --emit-ref-confidence GVCF && rm -f \",\"~/data/TRD/\",cross,\".runningGATK\"))\n",
    "\n",
    "    sbatch_list=execute_complex_sbatch(cmds, jobname = jobname<-\"GATK\", scripts_dir = scripts_dir, uniqueRunID = cross, cores=\"1\", mem=\"32gb\", time=\"long\", env=\"bwaetc\", initial_timedate = initial_timedate, jobs_simul = 10, jobs_total = 10)\n",
    "\n",
    "\n",
    "}\n",
    "\n",
    "if(exists(\"sbatch_list\")){\n",
    "    print(\"Batching up...\")\n",
    "    start_sbatch_list(sbatch_list, 10, jobname, initial_timedate)\n",
    "    }\n",
    "\n",
    "while(length(list.files(path = \"~/data/TRD\", pattern = \".runningGATK$\"))>0){\n",
    "    Sys.sleep(60)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46189d4f-58fb-49a4-9389-ac2660dc4a6c",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "library(readxl)\n",
    "\n",
    "crosses=list.files(\"~/data/trd/mapped_reads\",pattern = \"g.vcf.gz$\",full.names = TRUE)\n",
    "\n",
    "my_crosses=read_xlsx(\"~/data/trd/Crosses.xlsx\", sheet=2)[,c(\"Short name 1\", \"Short name 2\", \"Status\", \"Colonies in pool\")]\n",
    "chris_crosses=data.frame(sample1=c(\"ACP\",\"BAP\",\"CCD\",\"ATE\",\"ACK\",\"AKE\",\"BAH\",\"ANG\"),\n",
    "                        sample2=c(\"BFP\",\"CMP\",\"CPG\",\"YCR\",\"ACV\",\"BAH\",\"CGD\",\"CEI\"))\n",
    "all_used_samples=unique(c(my_crosses$`Short name 1`, my_crosses$`Short name 2`, chris_crosses$sample1, chris_crosses$sample2))\n",
    "all_used_samples[all_used_samples==\"YCR\"]=paste0(\"SACE_\",all_used_samples[all_used_samples==\"YCR\"])\n",
    "summary(file.exists(paste0(\"/home/jnrunge/data/trd/GVCF_2489Strains/\",all_used_samples,\".g.vcf.gz\")))\n",
    "\n",
    "all_used_samples[!file.exists(paste0(\"/home/jnrunge/data/trd/GVCF_2489Strains/\",all_used_samples,\".g.vcf.gz\"))]\n",
    "\n",
    "\n",
    "samples=c(crosses,paste0(\"/home/jnrunge/data/trd/GVCF_2489Strains/\",all_used_samples,\".g.vcf.gz\"))\n",
    "samples=data.frame(id=unlist(lapply(basename(samples), getFirst_v2, split=\".\")),file=samples)\n",
    "fwrite(samples, \"~/data/trd/mapped_reads/all_samples.tsv\", col.names = FALSE, sep=\"\\t\")\n",
    "head(samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3419da66-d987-4b20-ac02-c5985c661463",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# make BED file (all positions because GATK is retarded)\n",
    "fai=fread(\"~/data/TRD/R64_nucl.fasta.fai\")\n",
    "bed=data.frame(chr=fai$V1, from=1, to=fai$V2)\n",
    "head(bed)\n",
    "fwrite(bed, \"~/data/TRD/R64_nucl.fasta.fai.bed\", col.names = FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed983020-c77e-4a1e-9e15-9ed69401585a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "if(file.mtime(\"~/data/trd/mapped_reads/TRD.vcf.gz\")<max(file.mtime(samples$file))){\n",
    "    # if new samples can be added to the big vcf, then merge again\n",
    "    \n",
    "    file.create(running_file<-paste0(\"~/data/TRD/runningGATKmerge\"))\n",
    "    cmd=\"cd ~/data/trd/mapped_reads && rm -rf TRD_DB\"\n",
    "    cmd=paste0(cmd, \" && gatk GenomicsDBImport --batch-size 200 --genomicsdb-workspace-path TRD_DB --sample-name-map ~/data/trd/mapped_reads/all_samples.tsv -L ~/data/TRD/R64_nucl.fasta.fai.bed\")\n",
    "    cmd=paste0(cmd, \" && gatk GenotypeGVCFs -R ../../TRD/R64_nucl.fasta -V gendb://TRD_DB -O TRD.vcf.gz\")\n",
    "    cmd=paste0(cmd, \" && bcftools query -l TRD.vcf.gz > TRD.vcf.gz.samples && rm -f ~/data/TRD/runningGATKmerge\")\n",
    "    execute_cmd_sbatch(cmd, mem=\"32G\", cpu=\"1\", time=\"long\", env=\"bwaetc\", jobname=\"GATK_merge\")\n",
    "    \n",
    "    while(file.exists(running_file)){\n",
    "    Sys.sleep(60)\n",
    "}\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ca6c6cc-280a-4fa5-8b52-2bb5a92a6bc1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "Crosses=readxl::read_xlsx(\"~/data/trd/Crosses.xlsx\", sheet=2)\n",
    "head(Crosses<-Crosses[,c(\"Cross ID\",\"Short name 1\",\"Short name 2\")])\n",
    "tail(Crosses)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "186af08f-5c60-4f87-bde2-0641bdffb174",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "gvcf=\"/home/jnrunge/data/trd/mapped_reads/TRD.vcf.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d840fd15-d280-4651-a1c1-fca7ddbb5704",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ChrisC1\"\n",
      "[1] \"ChrisC2\"\n",
      "[1] \"ChrisC3\"\n",
      "[1] \"ChrisC4\"\n",
      "[1] \"ChrisC5\"\n",
      "[1] \"ChrisC6\"\n",
      "[1] \"ChrisC7\"\n",
      "[1] \"ChrisC8\"\n",
      "[1] \"YJNRC2\"\n",
      "[1] \"YJNRC17\"\n",
      "[1] \"YJNRC18\"\n"
     ]
    }
   ],
   "source": [
    "crosses=unlist(lapply(basename(bam_files), getFirst_v2, split=\".\"))\n",
    "\n",
    "for(cross in crosses){\n",
    "    run=FALSE\n",
    "    print(cross)\n",
    "    gvcf<-paste0(\"/home/jnrunge/data/trd/mapped_reads/\",cross,\".g.vcf.gz\")\n",
    "    if(file.exists(ohfile<-paste0(\"~/data/trd/mapped_reads/\",cross,\".hetLoci.gz\"))){\n",
    "        if(file.mtime(gvcf)>file.mtime(ohfile)){\n",
    "            run=TRUE\n",
    "            \n",
    "        }\n",
    "    }else{\n",
    "        run=TRUE\n",
    "    }\n",
    "    if(run == TRUE){\n",
    "            file.create(running_file<-paste0(\"~/data/TRD/Oppo-Homo-Pos/\",cross,\".runningOHLOCI\"))\n",
    "            cmd=paste0(\"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci2.sh \",cross)\n",
    "            cmd=paste0(cmd,\" && rm -f \",running_file)\n",
    "            execute_cmd_sbatch(cmd, mem=\"8G\", cpu=\"1\", time=\"short\", env=\"bwaetc\", jobname = \"OHLoci\")\n",
    "        Sys.sleep(1)\n",
    "            print(cmd)\n",
    "    }\n",
    "    }\n",
    "while(length(list.files(path = \"~/data/TRD/Oppo-Homo-Pos\", pattern = \".runningOHLOCI$\"))>0){\n",
    "    Sys.sleep(60)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "247ce53d-af7a-4d09-a291-c228d71c5740",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"BAK\"\n",
      "[1] \"BAN\"\n",
      "[1] \"AKH\"\n",
      "[1] \"ABS\"\n",
      "[1] \"AMM\"\n",
      "[1] \"ABE\"\n",
      "[1] \"BFC\"\n",
      "[1] \"BID\"\n",
      "[1] \"ACT\"\n",
      "[1] \"AKV\"\n",
      "[1] \"AMD\"\n",
      "[1] \"ANI\"\n",
      "[1] \"AND\"\n",
      "[1] \"ANH\"\n",
      "[1] \"AVI\"\n",
      "[1] \"ACI\"\n",
      "[1] \"ABC\"\n",
      "[1] \"ACN\"\n",
      "[1] \"AAM\"\n",
      "[1] \"ABG\"\n",
      "[1] \"AAR\"\n",
      "[1] \"ABA\"\n",
      "[1] \"BEF\"\n",
      "[1] \"AKI\"\n",
      "[1] \"ATE\"\n",
      "[1] \"BET\"\n",
      "[1] \"BTI\"\n",
      "[1] \"BQC\"\n",
      "[1] \"BMK\"\n",
      "[1] \"CRB\"\n",
      "[1] \"BKL\"\n",
      "[1] \"ABL\"\n",
      "[1] \"CCF\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CCF && rm -f ~/data/TRD/Oppo-Homo-Pos/CCF.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484897\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CCF && rm -f ~/data/TRD/Oppo-Homo-Pos/CCF.runningOHLOCI\"\n",
      "[1] \"CCC\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CCC && rm -f ~/data/TRD/Oppo-Homo-Pos/CCC.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484898\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CCC && rm -f ~/data/TRD/Oppo-Homo-Pos/CCC.runningOHLOCI\"\n",
      "[1] \"CCG\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CCG && rm -f ~/data/TRD/Oppo-Homo-Pos/CCG.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484899\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CCG && rm -f ~/data/TRD/Oppo-Homo-Pos/CCG.runningOHLOCI\"\n",
      "[1] \"CMQ\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CMQ && rm -f ~/data/TRD/Oppo-Homo-Pos/CMQ.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484900\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CMQ && rm -f ~/data/TRD/Oppo-Homo-Pos/CMQ.runningOHLOCI\"\n",
      "[1] \"BFQ\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BFQ && rm -f ~/data/TRD/Oppo-Homo-Pos/BFQ.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484901\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BFQ && rm -f ~/data/TRD/Oppo-Homo-Pos/BFQ.runningOHLOCI\"\n",
      "[1] \"BFR\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BFR && rm -f ~/data/TRD/Oppo-Homo-Pos/BFR.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484902\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BFR && rm -f ~/data/TRD/Oppo-Homo-Pos/BFR.runningOHLOCI\"\n",
      "[1] \"BHN\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BHN && rm -f ~/data/TRD/Oppo-Homo-Pos/BHN.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484903\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BHN && rm -f ~/data/TRD/Oppo-Homo-Pos/BHN.runningOHLOCI\"\n",
      "[1] \"BFD\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BFD && rm -f ~/data/TRD/Oppo-Homo-Pos/BFD.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484904\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BFD && rm -f ~/data/TRD/Oppo-Homo-Pos/BFD.runningOHLOCI\"\n",
      "[1] \"ABP\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ABP && rm -f ~/data/TRD/Oppo-Homo-Pos/ABP.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484905\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ABP && rm -f ~/data/TRD/Oppo-Homo-Pos/ABP.runningOHLOCI\"\n",
      "[1] \"AKS\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh AKS && rm -f ~/data/TRD/Oppo-Homo-Pos/AKS.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484906\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh AKS && rm -f ~/data/TRD/Oppo-Homo-Pos/AKS.runningOHLOCI\"\n",
      "[1] \"AKQ\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh AKQ && rm -f ~/data/TRD/Oppo-Homo-Pos/AKQ.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484907\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh AKQ && rm -f ~/data/TRD/Oppo-Homo-Pos/AKQ.runningOHLOCI\"\n",
      "[1] \"CPG\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CPG && rm -f ~/data/TRD/Oppo-Homo-Pos/CPG.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484908\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CPG && rm -f ~/data/TRD/Oppo-Homo-Pos/CPG.runningOHLOCI\"\n",
      "[1] \"ACP\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ACP && rm -f ~/data/TRD/Oppo-Homo-Pos/ACP.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484909\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ACP && rm -f ~/data/TRD/Oppo-Homo-Pos/ACP.runningOHLOCI\"\n",
      "[1] \"BAP\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BAP && rm -f ~/data/TRD/Oppo-Homo-Pos/BAP.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484910\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BAP && rm -f ~/data/TRD/Oppo-Homo-Pos/BAP.runningOHLOCI\"\n",
      "[1] \"CCD\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CCD && rm -f ~/data/TRD/Oppo-Homo-Pos/CCD.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484911\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CCD && rm -f ~/data/TRD/Oppo-Homo-Pos/CCD.runningOHLOCI\"\n",
      "[1] \"ACK\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ACK && rm -f ~/data/TRD/Oppo-Homo-Pos/ACK.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484912\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ACK && rm -f ~/data/TRD/Oppo-Homo-Pos/ACK.runningOHLOCI\"\n",
      "[1] \"AKE\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh AKE && rm -f ~/data/TRD/Oppo-Homo-Pos/AKE.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484913\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh AKE && rm -f ~/data/TRD/Oppo-Homo-Pos/AKE.runningOHLOCI\"\n",
      "[1] \"BAH\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BAH && rm -f ~/data/TRD/Oppo-Homo-Pos/BAH.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484914\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BAH && rm -f ~/data/TRD/Oppo-Homo-Pos/BAH.runningOHLOCI\"\n",
      "[1] \"ANG\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ANG && rm -f ~/data/TRD/Oppo-Homo-Pos/ANG.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484915\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ANG && rm -f ~/data/TRD/Oppo-Homo-Pos/ANG.runningOHLOCI\"\n",
      "[1] \"BFP\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BFP && rm -f ~/data/TRD/Oppo-Homo-Pos/BFP.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484916\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh BFP && rm -f ~/data/TRD/Oppo-Homo-Pos/BFP.runningOHLOCI\"\n",
      "[1] \"CMP\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CMP && rm -f ~/data/TRD/Oppo-Homo-Pos/CMP.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484917\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CMP && rm -f ~/data/TRD/Oppo-Homo-Pos/CMP.runningOHLOCI\"\n",
      "[1] \"SACE_YCR\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh SACE_YCR && rm -f ~/data/TRD/Oppo-Homo-Pos/SACE_YCR.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484918\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh SACE_YCR && rm -f ~/data/TRD/Oppo-Homo-Pos/SACE_YCR.runningOHLOCI\"\n",
      "[1] \"ACV\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ACV && rm -f ~/data/TRD/Oppo-Homo-Pos/ACV.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484919\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh ACV && rm -f ~/data/TRD/Oppo-Homo-Pos/ACV.runningOHLOCI\"\n",
      "[1] \"CGD\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CGD && rm -f ~/data/TRD/Oppo-Homo-Pos/CGD.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484920\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CGD && rm -f ~/data/TRD/Oppo-Homo-Pos/CGD.runningOHLOCI\"\n",
      "[1] \"CEI\"\n",
      "[1] \"sbatch -c 1 --mem=8G --job-name=OHLoci -p fast -t 11:59:00 --wrap '. ~/activate.sh bwaetc; cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CEI && rm -f ~/data/TRD/Oppo-Homo-Pos/CEI.runningOHLOCI'\"\n",
      "[1] \"Submitted batch job 484921\"\n",
      "[1] \"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh CEI && rm -f ~/data/TRD/Oppo-Homo-Pos/CEI.runningOHLOCI\"\n"
     ]
    }
   ],
   "source": [
    "for(s in samples$id){\n",
    "    if(grepl(\"YJNRC\",s,fixed = TRUE)|grepl(\"Chris\",s,fixed = TRUE)){\n",
    "        next\n",
    "    }\n",
    "    run=FALSE\n",
    "    print(s)\n",
    "    if(file.exists(ohfile<-paste0(\"~/data/trd/mapped_reads/\",s,\".homLoci.gz\"))){\n",
    "        if(file.mtime(gvcf)>file.mtime(ohfile)){\n",
    "            run=TRUE\n",
    "            \n",
    "        }\n",
    "    }else{\n",
    "        run=TRUE\n",
    "    }\n",
    "    if(run == TRUE){\n",
    "            file.create(running_file<-paste0(\"~/data/TRD/Oppo-Homo-Pos/\",s,\".runningOHLOCI\"))\n",
    "            cmd=paste0(\"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh \",s)\n",
    "            cmd=paste0(cmd,\" && rm -f \",running_file)\n",
    "            execute_cmd_sbatch(cmd, mem=\"8G\", cpu=\"1\", time=\"short\", env=\"bwaetc\", jobname = \"OHLoci\")\n",
    "            print(cmd)\n",
    "        Sys.sleep(1)\n",
    "        }\n",
    "    }\n",
    "while(length(list.files(path = \"~/data/TRD/Oppo-Homo-Pos\", pattern = \".runningOHLOCI$\"))>0){\n",
    "    Sys.sleep(60)\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
