{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "70c6951e-ce10-48d3-a0af-3045b472fb84",
   "metadata": {},
   "source": [
    "The goal here is to joint-call variants across the genome using GATK. Even though we don't need variants per se, we want to use GATK's repositioning (INDELS) to have as comparable a data set as possible, then extract the allele frequencies for TRD calling."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "78ac94e6-bc0b-4462-8237-4c95ef2006bd",
   "metadata": {},
   "source": [
    "What I need to do here to make sure that there is no confusion about the positions of opposite homozygotes is to call SNPs using the joint calling with the 2K matrix."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7774fd01-b570-4167-a858-e92e4054f425",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ChrisC1\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC2\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC3\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC4\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC5\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC6\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC7\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"ChrisC8\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC1\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC2\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC3\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC4\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC5\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC6\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC7\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC8\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC9\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC11\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC12\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC14\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC15\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC16\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC17\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC18\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC19\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC20\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC21\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC22\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC23\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC24\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC25\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC26\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC27\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC29\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC30\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC31\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC32\"\n",
      "[1] \"GVCF exists\"\n",
      "[1] \"YJNRC34\"\n",
      "[1] \"GVCF exists\"\n"
     ]
    }
   ],
   "source": [
    "source(\"../../BrusselSprouts/scripts/functions.R\")\n",
    "\n",
    "\n",
    "scripts_dir=\"/home/jnrunge/data/trd/mapped_reads/scripts/\"\n",
    "\n",
    "initial_timedate=Sys.time()\n",
    "\n",
    "# what needs to be done is individually, and parallely genotype all crosses, then merge them into a DB, and once call the joint genotyping (once per batch of sequencing, so twice in total)\n",
    "\n",
    "bam_files <- list.files(\n",
    "  path = \"~/data/trd/mapped_reads\",\n",
    "  pattern = \"YJNRC[0-9]*.bam$\",\n",
    "  full.names = TRUE\n",
    ") %>% \n",
    "  grep(pattern = \"dontuse\", invert = TRUE, value = TRUE)\n",
    "\n",
    "\n",
    "bam_files=c(bam_files, list.files(path=\"~/data/trd/mapped_reads\",pattern=\"ChrisC[0-9].bam$\",full.names = TRUE))\n",
    "bam_files=naturalsort(bam_files)\n",
    "crosses=unlist(lapply(basename(bam_files), getFirst_v2, split=\".\"))\n",
    "\n",
    "for(cross in crosses){\n",
    "    print(cross)\n",
    "\n",
    "    if(file.exists(paste0(\"/home/jnrunge/data/trd/mapped_reads/\",cross,\".g.vcf.gz\"))){\n",
    "        if(file.mtime(paste0(\"/home/jnrunge/data/trd/mapped_reads/\",cross,\".g.vcf.gz\"))>file.mtime(file.path(\"/home/jnrunge/data/trd/mapped_reads/\",paste0(cross,\".bam\")))){\n",
    "            print(\"GVCF exists\")\n",
    "            next\n",
    "        }\n",
    "    }\n",
    "    stop()\n",
    "    file.create(running_file<-paste0(\"~/data/TRD/\",cross,\".runningGATK\"))\n",
    "    cmds=paste0(\"cd /home/jnrunge/data/trd/mapped_reads/ && gatk AddOrReplaceReadGroups -I \",cross,\".bam -O \",cross,\".ReadGroups.bam --RGID \",cross,\" --RGLB \",cross,\" --RGPL ILLUMINA --RGPU \",cross,\" --RGSM \",cross,\"\")\n",
    "    cmds=c(cmds,paste0(\"mv -f \",cross,\".ReadGroups.bam \",cross,\".bam\"))\n",
    "    cmds=c(cmds,paste0(\"samtools index \",cross,\".bam\"))\n",
    "    cmds=c(cmds,paste0(\"gatk HaplotypeCaller -R ../../TRD/R64_nucl.fasta -I \",cross,\".bam -O \",cross,\".g.vcf.gz --emit-ref-confidence GVCF && rm -f \",\"~/data/TRD/\",cross,\".runningGATK\"))\n",
    "\n",
    "    sbatch_list=execute_complex_sbatch(cmds, jobname = jobname<-\"GATK\", scripts_dir = scripts_dir, uniqueRunID = cross, cores=\"1\", mem=\"32gb\", time=\"long\", env=\"bwaetc\", initial_timedate = initial_timedate, jobs_simul = 10, jobs_total = 20)\n",
    "\n",
    "\n",
    "}\n",
    "\n",
    "if(exists(\"sbatch_list\")){\n",
    "    print(\"Batching up...\")\n",
    "    start_sbatch_list(sbatch_list, 10, jobname, initial_timedate)\n",
    "    }\n",
    "\n",
    "while(length(list.files(path = \"~/data/TRD\", pattern = \".runningGATK$\"))>0){\n",
    "    Sys.sleep(60)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "46189d4f-58fb-49a4-9389-ac2660dc4a6c",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m*\u001b[39m `Is in Stock` -> `Is in Stock...4`\n",
      "\u001b[36m*\u001b[39m `Is in Stock` -> `Is in Stock...6`\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "   Mode    TRUE \n",
       "logical      57 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 95 x 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>id</th><th scope=col>file</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ChrisC1</td><td>/home/jnrunge/data/trd/mapped_reads/ChrisC1.g.vcf.gz</td></tr>\n",
       "\t<tr><td>ChrisC2</td><td>/home/jnrunge/data/trd/mapped_reads/ChrisC2.g.vcf.gz</td></tr>\n",
       "\t<tr><td>ChrisC3</td><td>/home/jnrunge/data/trd/mapped_reads/ChrisC3.g.vcf.gz</td></tr>\n",
       "\t<tr><td>ChrisC4</td><td>/home/jnrunge/data/trd/mapped_reads/ChrisC4.g.vcf.gz</td></tr>\n",
       "\t<tr><td>ChrisC5</td><td>/home/jnrunge/data/trd/mapped_reads/ChrisC5.g.vcf.gz</td></tr>\n",
       "\t<tr><td>ChrisC6</td><td>/home/jnrunge/data/trd/mapped_reads/ChrisC6.g.vcf.gz</td></tr>\n",
       "\t<tr><td>ChrisC7</td><td>/home/jnrunge/data/trd/mapped_reads/ChrisC7.g.vcf.gz</td></tr>\n",
       "\t<tr><td>ChrisC8</td><td>/home/jnrunge/data/trd/mapped_reads/ChrisC8.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC1 </td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC1.g.vcf.gz </td></tr>\n",
       "\t<tr><td>YJNRC11</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC11.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC12</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC12.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC14</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC14.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC15</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC15.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC16</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC16.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC17</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC17.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC18</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC18.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC19</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC19.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC2 </td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC2.g.vcf.gz </td></tr>\n",
       "\t<tr><td>YJNRC20</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC20.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC21</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC21.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC22</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC22.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC23</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC23.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC24</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC24.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC25</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC25.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC26</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC26.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC27</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC27.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC29</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC29.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC3 </td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC3.g.vcf.gz </td></tr>\n",
       "\t<tr><td>YJNRC30</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC30.g.vcf.gz</td></tr>\n",
       "\t<tr><td>YJNRC31</td><td>/home/jnrunge/data/trd/mapped_reads/YJNRC31.g.vcf.gz</td></tr>\n",
       "\t<tr><td>...</td><td>...</td></tr>\n",
       "\t<tr><td>BQC     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BQC.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>BMK     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BMK.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CRB     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CRB.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>BKL     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BKL.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>ABL     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/ABL.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CCF     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CCF.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CCC     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CCC.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CCG     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CCG.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CMQ     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CMQ.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>BFQ     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BFQ.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>BFR     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BFR.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>BHN     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BHN.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>BFD     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BFD.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>ABP     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/ABP.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>AKS     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/AKS.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>AKQ     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/AKQ.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CPG     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CPG.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>ACP     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/ACP.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>BAP     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BAP.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CCD     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CCD.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>ACK     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/ACK.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>AKE     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/AKE.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>BAH     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BAH.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>ANG     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/ANG.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>BFP     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/BFP.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CMP     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CMP.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>SACE_YCR</td><td>/home/jnrunge/data/trd/GVCF_2489Strains/SACE_YCR.g.vcf.gz</td></tr>\n",
       "\t<tr><td>ACV     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/ACV.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CGD     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CGD.g.vcf.gz     </td></tr>\n",
       "\t<tr><td>CEI     </td><td>/home/jnrunge/data/trd/GVCF_2489Strains/CEI.g.vcf.gz     </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 95 x 2\n",
       "\\begin{tabular}{ll}\n",
       " id & file\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ChrisC1 & /home/jnrunge/data/trd/mapped\\_reads/ChrisC1.g.vcf.gz\\\\\n",
       "\t ChrisC2 & /home/jnrunge/data/trd/mapped\\_reads/ChrisC2.g.vcf.gz\\\\\n",
       "\t ChrisC3 & /home/jnrunge/data/trd/mapped\\_reads/ChrisC3.g.vcf.gz\\\\\n",
       "\t ChrisC4 & /home/jnrunge/data/trd/mapped\\_reads/ChrisC4.g.vcf.gz\\\\\n",
       "\t ChrisC5 & /home/jnrunge/data/trd/mapped\\_reads/ChrisC5.g.vcf.gz\\\\\n",
       "\t ChrisC6 & /home/jnrunge/data/trd/mapped\\_reads/ChrisC6.g.vcf.gz\\\\\n",
       "\t ChrisC7 & /home/jnrunge/data/trd/mapped\\_reads/ChrisC7.g.vcf.gz\\\\\n",
       "\t ChrisC8 & /home/jnrunge/data/trd/mapped\\_reads/ChrisC8.g.vcf.gz\\\\\n",
       "\t YJNRC1  & /home/jnrunge/data/trd/mapped\\_reads/YJNRC1.g.vcf.gz \\\\\n",
       "\t YJNRC11 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC11.g.vcf.gz\\\\\n",
       "\t YJNRC12 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC12.g.vcf.gz\\\\\n",
       "\t YJNRC14 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC14.g.vcf.gz\\\\\n",
       "\t YJNRC15 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC15.g.vcf.gz\\\\\n",
       "\t YJNRC16 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC16.g.vcf.gz\\\\\n",
       "\t YJNRC17 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC17.g.vcf.gz\\\\\n",
       "\t YJNRC18 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC18.g.vcf.gz\\\\\n",
       "\t YJNRC19 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC19.g.vcf.gz\\\\\n",
       "\t YJNRC2  & /home/jnrunge/data/trd/mapped\\_reads/YJNRC2.g.vcf.gz \\\\\n",
       "\t YJNRC20 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC20.g.vcf.gz\\\\\n",
       "\t YJNRC21 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC21.g.vcf.gz\\\\\n",
       "\t YJNRC22 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC22.g.vcf.gz\\\\\n",
       "\t YJNRC23 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC23.g.vcf.gz\\\\\n",
       "\t YJNRC24 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC24.g.vcf.gz\\\\\n",
       "\t YJNRC25 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC25.g.vcf.gz\\\\\n",
       "\t YJNRC26 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC26.g.vcf.gz\\\\\n",
       "\t YJNRC27 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC27.g.vcf.gz\\\\\n",
       "\t YJNRC29 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC29.g.vcf.gz\\\\\n",
       "\t YJNRC3  & /home/jnrunge/data/trd/mapped\\_reads/YJNRC3.g.vcf.gz \\\\\n",
       "\t YJNRC30 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC30.g.vcf.gz\\\\\n",
       "\t YJNRC31 & /home/jnrunge/data/trd/mapped\\_reads/YJNRC31.g.vcf.gz\\\\\n",
       "\t ... & ...\\\\\n",
       "\t BQC      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BQC.g.vcf.gz     \\\\\n",
       "\t BMK      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BMK.g.vcf.gz     \\\\\n",
       "\t CRB      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CRB.g.vcf.gz     \\\\\n",
       "\t BKL      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BKL.g.vcf.gz     \\\\\n",
       "\t ABL      & /home/jnrunge/data/trd/GVCF\\_2489Strains/ABL.g.vcf.gz     \\\\\n",
       "\t CCF      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CCF.g.vcf.gz     \\\\\n",
       "\t CCC      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CCC.g.vcf.gz     \\\\\n",
       "\t CCG      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CCG.g.vcf.gz     \\\\\n",
       "\t CMQ      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CMQ.g.vcf.gz     \\\\\n",
       "\t BFQ      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BFQ.g.vcf.gz     \\\\\n",
       "\t BFR      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BFR.g.vcf.gz     \\\\\n",
       "\t BHN      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BHN.g.vcf.gz     \\\\\n",
       "\t BFD      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BFD.g.vcf.gz     \\\\\n",
       "\t ABP      & /home/jnrunge/data/trd/GVCF\\_2489Strains/ABP.g.vcf.gz     \\\\\n",
       "\t AKS      & /home/jnrunge/data/trd/GVCF\\_2489Strains/AKS.g.vcf.gz     \\\\\n",
       "\t AKQ      & /home/jnrunge/data/trd/GVCF\\_2489Strains/AKQ.g.vcf.gz     \\\\\n",
       "\t CPG      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CPG.g.vcf.gz     \\\\\n",
       "\t ACP      & /home/jnrunge/data/trd/GVCF\\_2489Strains/ACP.g.vcf.gz     \\\\\n",
       "\t BAP      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BAP.g.vcf.gz     \\\\\n",
       "\t CCD      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CCD.g.vcf.gz     \\\\\n",
       "\t ACK      & /home/jnrunge/data/trd/GVCF\\_2489Strains/ACK.g.vcf.gz     \\\\\n",
       "\t AKE      & /home/jnrunge/data/trd/GVCF\\_2489Strains/AKE.g.vcf.gz     \\\\\n",
       "\t BAH      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BAH.g.vcf.gz     \\\\\n",
       "\t ANG      & /home/jnrunge/data/trd/GVCF\\_2489Strains/ANG.g.vcf.gz     \\\\\n",
       "\t BFP      & /home/jnrunge/data/trd/GVCF\\_2489Strains/BFP.g.vcf.gz     \\\\\n",
       "\t CMP      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CMP.g.vcf.gz     \\\\\n",
       "\t SACE\\_YCR & /home/jnrunge/data/trd/GVCF\\_2489Strains/SACE\\_YCR.g.vcf.gz\\\\\n",
       "\t ACV      & /home/jnrunge/data/trd/GVCF\\_2489Strains/ACV.g.vcf.gz     \\\\\n",
       "\t CGD      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CGD.g.vcf.gz     \\\\\n",
       "\t CEI      & /home/jnrunge/data/trd/GVCF\\_2489Strains/CEI.g.vcf.gz     \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 95 x 2\n",
       "\n",
       "| id &lt;chr&gt; | file &lt;chr&gt; |\n",
       "|---|---|\n",
       "| ChrisC1 | /home/jnrunge/data/trd/mapped_reads/ChrisC1.g.vcf.gz |\n",
       "| ChrisC2 | /home/jnrunge/data/trd/mapped_reads/ChrisC2.g.vcf.gz |\n",
       "| ChrisC3 | /home/jnrunge/data/trd/mapped_reads/ChrisC3.g.vcf.gz |\n",
       "| ChrisC4 | /home/jnrunge/data/trd/mapped_reads/ChrisC4.g.vcf.gz |\n",
       "| ChrisC5 | /home/jnrunge/data/trd/mapped_reads/ChrisC5.g.vcf.gz |\n",
       "| ChrisC6 | /home/jnrunge/data/trd/mapped_reads/ChrisC6.g.vcf.gz |\n",
       "| ChrisC7 | /home/jnrunge/data/trd/mapped_reads/ChrisC7.g.vcf.gz |\n",
       "| ChrisC8 | /home/jnrunge/data/trd/mapped_reads/ChrisC8.g.vcf.gz |\n",
       "| YJNRC1  | /home/jnrunge/data/trd/mapped_reads/YJNRC1.g.vcf.gz  |\n",
       "| YJNRC11 | /home/jnrunge/data/trd/mapped_reads/YJNRC11.g.vcf.gz |\n",
       "| YJNRC12 | /home/jnrunge/data/trd/mapped_reads/YJNRC12.g.vcf.gz |\n",
       "| YJNRC14 | /home/jnrunge/data/trd/mapped_reads/YJNRC14.g.vcf.gz |\n",
       "| YJNRC15 | /home/jnrunge/data/trd/mapped_reads/YJNRC15.g.vcf.gz |\n",
       "| YJNRC16 | /home/jnrunge/data/trd/mapped_reads/YJNRC16.g.vcf.gz |\n",
       "| YJNRC17 | /home/jnrunge/data/trd/mapped_reads/YJNRC17.g.vcf.gz |\n",
       "| YJNRC18 | /home/jnrunge/data/trd/mapped_reads/YJNRC18.g.vcf.gz |\n",
       "| YJNRC19 | /home/jnrunge/data/trd/mapped_reads/YJNRC19.g.vcf.gz |\n",
       "| YJNRC2  | /home/jnrunge/data/trd/mapped_reads/YJNRC2.g.vcf.gz  |\n",
       "| YJNRC20 | /home/jnrunge/data/trd/mapped_reads/YJNRC20.g.vcf.gz |\n",
       "| YJNRC21 | /home/jnrunge/data/trd/mapped_reads/YJNRC21.g.vcf.gz |\n",
       "| YJNRC22 | /home/jnrunge/data/trd/mapped_reads/YJNRC22.g.vcf.gz |\n",
       "| YJNRC23 | /home/jnrunge/data/trd/mapped_reads/YJNRC23.g.vcf.gz |\n",
       "| YJNRC24 | /home/jnrunge/data/trd/mapped_reads/YJNRC24.g.vcf.gz |\n",
       "| YJNRC25 | /home/jnrunge/data/trd/mapped_reads/YJNRC25.g.vcf.gz |\n",
       "| YJNRC26 | /home/jnrunge/data/trd/mapped_reads/YJNRC26.g.vcf.gz |\n",
       "| YJNRC27 | /home/jnrunge/data/trd/mapped_reads/YJNRC27.g.vcf.gz |\n",
       "| YJNRC29 | /home/jnrunge/data/trd/mapped_reads/YJNRC29.g.vcf.gz |\n",
       "| YJNRC3  | /home/jnrunge/data/trd/mapped_reads/YJNRC3.g.vcf.gz  |\n",
       "| YJNRC30 | /home/jnrunge/data/trd/mapped_reads/YJNRC30.g.vcf.gz |\n",
       "| YJNRC31 | /home/jnrunge/data/trd/mapped_reads/YJNRC31.g.vcf.gz |\n",
       "| ... | ... |\n",
       "| BQC      | /home/jnrunge/data/trd/GVCF_2489Strains/BQC.g.vcf.gz      |\n",
       "| BMK      | /home/jnrunge/data/trd/GVCF_2489Strains/BMK.g.vcf.gz      |\n",
       "| CRB      | /home/jnrunge/data/trd/GVCF_2489Strains/CRB.g.vcf.gz      |\n",
       "| BKL      | /home/jnrunge/data/trd/GVCF_2489Strains/BKL.g.vcf.gz      |\n",
       "| ABL      | /home/jnrunge/data/trd/GVCF_2489Strains/ABL.g.vcf.gz      |\n",
       "| CCF      | /home/jnrunge/data/trd/GVCF_2489Strains/CCF.g.vcf.gz      |\n",
       "| CCC      | /home/jnrunge/data/trd/GVCF_2489Strains/CCC.g.vcf.gz      |\n",
       "| CCG      | /home/jnrunge/data/trd/GVCF_2489Strains/CCG.g.vcf.gz      |\n",
       "| CMQ      | /home/jnrunge/data/trd/GVCF_2489Strains/CMQ.g.vcf.gz      |\n",
       "| BFQ      | /home/jnrunge/data/trd/GVCF_2489Strains/BFQ.g.vcf.gz      |\n",
       "| BFR      | /home/jnrunge/data/trd/GVCF_2489Strains/BFR.g.vcf.gz      |\n",
       "| BHN      | /home/jnrunge/data/trd/GVCF_2489Strains/BHN.g.vcf.gz      |\n",
       "| BFD      | /home/jnrunge/data/trd/GVCF_2489Strains/BFD.g.vcf.gz      |\n",
       "| ABP      | /home/jnrunge/data/trd/GVCF_2489Strains/ABP.g.vcf.gz      |\n",
       "| AKS      | /home/jnrunge/data/trd/GVCF_2489Strains/AKS.g.vcf.gz      |\n",
       "| AKQ      | /home/jnrunge/data/trd/GVCF_2489Strains/AKQ.g.vcf.gz      |\n",
       "| CPG      | /home/jnrunge/data/trd/GVCF_2489Strains/CPG.g.vcf.gz      |\n",
       "| ACP      | /home/jnrunge/data/trd/GVCF_2489Strains/ACP.g.vcf.gz      |\n",
       "| BAP      | /home/jnrunge/data/trd/GVCF_2489Strains/BAP.g.vcf.gz      |\n",
       "| CCD      | /home/jnrunge/data/trd/GVCF_2489Strains/CCD.g.vcf.gz      |\n",
       "| ACK      | /home/jnrunge/data/trd/GVCF_2489Strains/ACK.g.vcf.gz      |\n",
       "| AKE      | /home/jnrunge/data/trd/GVCF_2489Strains/AKE.g.vcf.gz      |\n",
       "| BAH      | /home/jnrunge/data/trd/GVCF_2489Strains/BAH.g.vcf.gz      |\n",
       "| ANG      | /home/jnrunge/data/trd/GVCF_2489Strains/ANG.g.vcf.gz      |\n",
       "| BFP      | /home/jnrunge/data/trd/GVCF_2489Strains/BFP.g.vcf.gz      |\n",
       "| CMP      | /home/jnrunge/data/trd/GVCF_2489Strains/CMP.g.vcf.gz      |\n",
       "| SACE_YCR | /home/jnrunge/data/trd/GVCF_2489Strains/SACE_YCR.g.vcf.gz |\n",
       "| ACV      | /home/jnrunge/data/trd/GVCF_2489Strains/ACV.g.vcf.gz      |\n",
       "| CGD      | /home/jnrunge/data/trd/GVCF_2489Strains/CGD.g.vcf.gz      |\n",
       "| CEI      | /home/jnrunge/data/trd/GVCF_2489Strains/CEI.g.vcf.gz      |\n",
       "\n"
      ],
      "text/plain": [
       "    id       file                                                     \n",
       "1   ChrisC1  /home/jnrunge/data/trd/mapped_reads/ChrisC1.g.vcf.gz     \n",
       "2   ChrisC2  /home/jnrunge/data/trd/mapped_reads/ChrisC2.g.vcf.gz     \n",
       "3   ChrisC3  /home/jnrunge/data/trd/mapped_reads/ChrisC3.g.vcf.gz     \n",
       "4   ChrisC4  /home/jnrunge/data/trd/mapped_reads/ChrisC4.g.vcf.gz     \n",
       "5   ChrisC5  /home/jnrunge/data/trd/mapped_reads/ChrisC5.g.vcf.gz     \n",
       "6   ChrisC6  /home/jnrunge/data/trd/mapped_reads/ChrisC6.g.vcf.gz     \n",
       "7   ChrisC7  /home/jnrunge/data/trd/mapped_reads/ChrisC7.g.vcf.gz     \n",
       "8   ChrisC8  /home/jnrunge/data/trd/mapped_reads/ChrisC8.g.vcf.gz     \n",
       "9   YJNRC1   /home/jnrunge/data/trd/mapped_reads/YJNRC1.g.vcf.gz      \n",
       "10  YJNRC11  /home/jnrunge/data/trd/mapped_reads/YJNRC11.g.vcf.gz     \n",
       "11  YJNRC12  /home/jnrunge/data/trd/mapped_reads/YJNRC12.g.vcf.gz     \n",
       "12  YJNRC14  /home/jnrunge/data/trd/mapped_reads/YJNRC14.g.vcf.gz     \n",
       "13  YJNRC15  /home/jnrunge/data/trd/mapped_reads/YJNRC15.g.vcf.gz     \n",
       "14  YJNRC16  /home/jnrunge/data/trd/mapped_reads/YJNRC16.g.vcf.gz     \n",
       "15  YJNRC17  /home/jnrunge/data/trd/mapped_reads/YJNRC17.g.vcf.gz     \n",
       "16  YJNRC18  /home/jnrunge/data/trd/mapped_reads/YJNRC18.g.vcf.gz     \n",
       "17  YJNRC19  /home/jnrunge/data/trd/mapped_reads/YJNRC19.g.vcf.gz     \n",
       "18  YJNRC2   /home/jnrunge/data/trd/mapped_reads/YJNRC2.g.vcf.gz      \n",
       "19  YJNRC20  /home/jnrunge/data/trd/mapped_reads/YJNRC20.g.vcf.gz     \n",
       "20  YJNRC21  /home/jnrunge/data/trd/mapped_reads/YJNRC21.g.vcf.gz     \n",
       "21  YJNRC22  /home/jnrunge/data/trd/mapped_reads/YJNRC22.g.vcf.gz     \n",
       "22  YJNRC23  /home/jnrunge/data/trd/mapped_reads/YJNRC23.g.vcf.gz     \n",
       "23  YJNRC24  /home/jnrunge/data/trd/mapped_reads/YJNRC24.g.vcf.gz     \n",
       "24  YJNRC25  /home/jnrunge/data/trd/mapped_reads/YJNRC25.g.vcf.gz     \n",
       "25  YJNRC26  /home/jnrunge/data/trd/mapped_reads/YJNRC26.g.vcf.gz     \n",
       "26  YJNRC27  /home/jnrunge/data/trd/mapped_reads/YJNRC27.g.vcf.gz     \n",
       "27  YJNRC29  /home/jnrunge/data/trd/mapped_reads/YJNRC29.g.vcf.gz     \n",
       "28  YJNRC3   /home/jnrunge/data/trd/mapped_reads/YJNRC3.g.vcf.gz      \n",
       "29  YJNRC30  /home/jnrunge/data/trd/mapped_reads/YJNRC30.g.vcf.gz     \n",
       "30  YJNRC31  /home/jnrunge/data/trd/mapped_reads/YJNRC31.g.vcf.gz     \n",
       "... ...      ...                                                      \n",
       "66  BQC      /home/jnrunge/data/trd/GVCF_2489Strains/BQC.g.vcf.gz     \n",
       "67  BMK      /home/jnrunge/data/trd/GVCF_2489Strains/BMK.g.vcf.gz     \n",
       "68  CRB      /home/jnrunge/data/trd/GVCF_2489Strains/CRB.g.vcf.gz     \n",
       "69  BKL      /home/jnrunge/data/trd/GVCF_2489Strains/BKL.g.vcf.gz     \n",
       "70  ABL      /home/jnrunge/data/trd/GVCF_2489Strains/ABL.g.vcf.gz     \n",
       "71  CCF      /home/jnrunge/data/trd/GVCF_2489Strains/CCF.g.vcf.gz     \n",
       "72  CCC      /home/jnrunge/data/trd/GVCF_2489Strains/CCC.g.vcf.gz     \n",
       "73  CCG      /home/jnrunge/data/trd/GVCF_2489Strains/CCG.g.vcf.gz     \n",
       "74  CMQ      /home/jnrunge/data/trd/GVCF_2489Strains/CMQ.g.vcf.gz     \n",
       "75  BFQ      /home/jnrunge/data/trd/GVCF_2489Strains/BFQ.g.vcf.gz     \n",
       "76  BFR      /home/jnrunge/data/trd/GVCF_2489Strains/BFR.g.vcf.gz     \n",
       "77  BHN      /home/jnrunge/data/trd/GVCF_2489Strains/BHN.g.vcf.gz     \n",
       "78  BFD      /home/jnrunge/data/trd/GVCF_2489Strains/BFD.g.vcf.gz     \n",
       "79  ABP      /home/jnrunge/data/trd/GVCF_2489Strains/ABP.g.vcf.gz     \n",
       "80  AKS      /home/jnrunge/data/trd/GVCF_2489Strains/AKS.g.vcf.gz     \n",
       "81  AKQ      /home/jnrunge/data/trd/GVCF_2489Strains/AKQ.g.vcf.gz     \n",
       "82  CPG      /home/jnrunge/data/trd/GVCF_2489Strains/CPG.g.vcf.gz     \n",
       "83  ACP      /home/jnrunge/data/trd/GVCF_2489Strains/ACP.g.vcf.gz     \n",
       "84  BAP      /home/jnrunge/data/trd/GVCF_2489Strains/BAP.g.vcf.gz     \n",
       "85  CCD      /home/jnrunge/data/trd/GVCF_2489Strains/CCD.g.vcf.gz     \n",
       "86  ACK      /home/jnrunge/data/trd/GVCF_2489Strains/ACK.g.vcf.gz     \n",
       "87  AKE      /home/jnrunge/data/trd/GVCF_2489Strains/AKE.g.vcf.gz     \n",
       "88  BAH      /home/jnrunge/data/trd/GVCF_2489Strains/BAH.g.vcf.gz     \n",
       "89  ANG      /home/jnrunge/data/trd/GVCF_2489Strains/ANG.g.vcf.gz     \n",
       "90  BFP      /home/jnrunge/data/trd/GVCF_2489Strains/BFP.g.vcf.gz     \n",
       "91  CMP      /home/jnrunge/data/trd/GVCF_2489Strains/CMP.g.vcf.gz     \n",
       "92  SACE_YCR /home/jnrunge/data/trd/GVCF_2489Strains/SACE_YCR.g.vcf.gz\n",
       "93  ACV      /home/jnrunge/data/trd/GVCF_2489Strains/ACV.g.vcf.gz     \n",
       "94  CGD      /home/jnrunge/data/trd/GVCF_2489Strains/CGD.g.vcf.gz     \n",
       "95  CEI      /home/jnrunge/data/trd/GVCF_2489Strains/CEI.g.vcf.gz     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(readxl)\n",
    "\n",
    "crosses=list.files(\"~/data/trd/mapped_reads\",pattern = \"g.vcf.gz$\",full.names = TRUE)\n",
    "\n",
    "my_crosses=read_xlsx(\"~/data/trd/Crosses.xlsx\", sheet=2)[,c(\"Short name 1\", \"Short name 2\", \"Status\", \"Colonies in pool\")]\n",
    "chris_crosses=data.frame(sample1=c(\"ACP\",\"BAP\",\"CCD\",\"ATE\",\"ACK\",\"AKE\",\"BAH\",\"ANG\"),\n",
    "                        sample2=c(\"BFP\",\"CMP\",\"CPG\",\"YCR\",\"ACV\",\"BAH\",\"CGD\",\"CEI\"))\n",
    "all_used_samples=unique(c(my_crosses$`Short name 1`, my_crosses$`Short name 2`, chris_crosses$sample1, chris_crosses$sample2))\n",
    "all_used_samples[all_used_samples==\"YCR\"]=paste0(\"SACE_\",all_used_samples[all_used_samples==\"YCR\"])\n",
    "summary(file.exists(paste0(\"/home/jnrunge/data/trd/GVCF_2489Strains/\",all_used_samples,\".g.vcf.gz\")))\n",
    "\n",
    "all_used_samples[!file.exists(paste0(\"/home/jnrunge/data/trd/GVCF_2489Strains/\",all_used_samples,\".g.vcf.gz\"))]\n",
    "\n",
    "\n",
    "samples=c(crosses,paste0(\"/home/jnrunge/data/trd/GVCF_2489Strains/\",all_used_samples,\".g.vcf.gz\"))\n",
    "samples=data.frame(id=unlist(lapply(basename(samples), getFirst_v2, split=\".\")),file=samples)\n",
    "fwrite(samples, \"~/data/trd/mapped_reads/all_samples.tsv\", col.names = FALSE, sep=\"\\t\")\n",
    "samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3419da66-d987-4b20-ac02-c5985c661463",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 x 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>from</th><th scope=col>to</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chromosome1</td><td>1</td><td> 230218</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chromosome2</td><td>1</td><td> 813184</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chromosome3</td><td>1</td><td> 316620</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chromosome4</td><td>1</td><td>1531933</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chromosome5</td><td>1</td><td> 576874</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chromosome6</td><td>1</td><td> 270161</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & chr & from & to\\\\\n",
       "  & <chr> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & chromosome1 & 1 &  230218\\\\\n",
       "\t2 & chromosome2 & 1 &  813184\\\\\n",
       "\t3 & chromosome3 & 1 &  316620\\\\\n",
       "\t4 & chromosome4 & 1 & 1531933\\\\\n",
       "\t5 & chromosome5 & 1 &  576874\\\\\n",
       "\t6 & chromosome6 & 1 &  270161\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 3\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | from &lt;dbl&gt; | to &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | chromosome1 | 1 |  230218 |\n",
       "| 2 | chromosome2 | 1 |  813184 |\n",
       "| 3 | chromosome3 | 1 |  316620 |\n",
       "| 4 | chromosome4 | 1 | 1531933 |\n",
       "| 5 | chromosome5 | 1 |  576874 |\n",
       "| 6 | chromosome6 | 1 |  270161 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr         from to     \n",
       "1 chromosome1 1     230218\n",
       "2 chromosome2 1     813184\n",
       "3 chromosome3 1     316620\n",
       "4 chromosome4 1    1531933\n",
       "5 chromosome5 1     576874\n",
       "6 chromosome6 1     270161"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# make BED file (all positions because GATK is retarded)\n",
    "fai=fread(\"~/data/TRD/R64_nucl.fasta.fai\")\n",
    "bed=data.frame(chr=fai$V1, from=1, to=fai$V2)\n",
    "head(bed)\n",
    "fwrite(bed, \"~/data/TRD/R64_nucl.fasta.fai.bed\", col.names = FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ed983020-c77e-4a1e-9e15-9ed69401585a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "if (file.mtime(\"~/data/trd/mapped_reads/TRD.vcf.gz\") < max(file.mtime(samples$file))) {\n",
    "  # if new samples can be added to the big vcf, then merge again\n",
    "\n",
    "  file.create(running_file <- paste0(\"~/data/TRD/runningGATKmerge\"))\n",
    "  cmd <- \"cd ~/data/trd/mapped_reads && rm -rf TRD_DB\"\n",
    "  cmd <- paste0(cmd, \" && gatk GenomicsDBImport --batch-size 200 --genomicsdb-workspace-path TRD_DB --sample-name-map ~/data/trd/mapped_reads/all_samples.tsv -L ~/data/TRD/R64_nucl.fasta.fai.bed\")\n",
    "  cmd <- paste0(cmd, \" && gatk GenotypeGVCFs -R ../../TRD/R64_nucl.fasta -V gendb://TRD_DB -O TRD.vcf.gz\")\n",
    "  cmd <- paste0(cmd, \" && bcftools query -l TRD.vcf.gz > TRD.vcf.gz.samples && rm -f ~/data/TRD/runningGATKmerge\")\n",
    "  execute_cmd_sbatch(cmd, mem = \"32G\", cpu = \"1\", time = \"long\", env = \"bwaetc\", jobname = \"GATK_merge\")\n",
    "\n",
    "  while (file.exists(running_file)) {\n",
    "    Sys.sleep(60)\n",
    "  }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3ca6c6cc-280a-4fa5-8b52-2bb5a92a6bc1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m*\u001b[39m `Is in Stock` -> `Is in Stock...4`\n",
      "\u001b[36m*\u001b[39m `Is in Stock` -> `Is in Stock...6`\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 x 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Cross ID</th><th scope=col>Short name 1</th><th scope=col>Short name 2</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>YJNRC1</td><td>BAK</td><td>BET</td></tr>\n",
       "\t<tr><td>YJNRC2</td><td>BAN</td><td>BTI</td></tr>\n",
       "\t<tr><td>YJNRC3</td><td>AKH</td><td>BQC</td></tr>\n",
       "\t<tr><td>YJNRC4</td><td>ABS</td><td>BEF</td></tr>\n",
       "\t<tr><td>YJNRC5</td><td>BAK</td><td>BMK</td></tr>\n",
       "\t<tr><td>YJNRC6</td><td>BAN</td><td>CRB</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 x 3\n",
       "\\begin{tabular}{lll}\n",
       " Cross ID & Short name 1 & Short name 2\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t YJNRC1 & BAK & BET\\\\\n",
       "\t YJNRC2 & BAN & BTI\\\\\n",
       "\t YJNRC3 & AKH & BQC\\\\\n",
       "\t YJNRC4 & ABS & BEF\\\\\n",
       "\t YJNRC5 & BAK & BMK\\\\\n",
       "\t YJNRC6 & BAN & CRB\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 x 3\n",
       "\n",
       "| Cross ID &lt;chr&gt; | Short name 1 &lt;chr&gt; | Short name 2 &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| YJNRC1 | BAK | BET |\n",
       "| YJNRC2 | BAN | BTI |\n",
       "| YJNRC3 | AKH | BQC |\n",
       "| YJNRC4 | ABS | BEF |\n",
       "| YJNRC5 | BAK | BMK |\n",
       "| YJNRC6 | BAN | CRB |\n",
       "\n"
      ],
      "text/plain": [
       "  Cross ID Short name 1 Short name 2\n",
       "1 YJNRC1   BAK          BET         \n",
       "2 YJNRC2   BAN          BTI         \n",
       "3 YJNRC3   AKH          BQC         \n",
       "4 YJNRC4   ABS          BEF         \n",
       "5 YJNRC5   BAK          BMK         \n",
       "6 YJNRC6   BAN          CRB         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 x 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Cross ID</th><th scope=col>Short name 1</th><th scope=col>Short name 2</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>YJNRC29</td><td>AAR</td><td>CPG</td></tr>\n",
       "\t<tr><td>YJNRC30</td><td>ABG</td><td>CPG</td></tr>\n",
       "\t<tr><td>YJNRC31</td><td>AKI</td><td>CPG</td></tr>\n",
       "\t<tr><td>YJNRC32</td><td>ATE</td><td>CPG</td></tr>\n",
       "\t<tr><td>YJNRC33</td><td>AAR</td><td>AKH</td></tr>\n",
       "\t<tr><td>YJNRC34</td><td>AKH</td><td>AKQ</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 x 3\n",
       "\\begin{tabular}{lll}\n",
       " Cross ID & Short name 1 & Short name 2\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t YJNRC29 & AAR & CPG\\\\\n",
       "\t YJNRC30 & ABG & CPG\\\\\n",
       "\t YJNRC31 & AKI & CPG\\\\\n",
       "\t YJNRC32 & ATE & CPG\\\\\n",
       "\t YJNRC33 & AAR & AKH\\\\\n",
       "\t YJNRC34 & AKH & AKQ\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 x 3\n",
       "\n",
       "| Cross ID &lt;chr&gt; | Short name 1 &lt;chr&gt; | Short name 2 &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| YJNRC29 | AAR | CPG |\n",
       "| YJNRC30 | ABG | CPG |\n",
       "| YJNRC31 | AKI | CPG |\n",
       "| YJNRC32 | ATE | CPG |\n",
       "| YJNRC33 | AAR | AKH |\n",
       "| YJNRC34 | AKH | AKQ |\n",
       "\n"
      ],
      "text/plain": [
       "  Cross ID Short name 1 Short name 2\n",
       "1 YJNRC29  AAR          CPG         \n",
       "2 YJNRC30  ABG          CPG         \n",
       "3 YJNRC31  AKI          CPG         \n",
       "4 YJNRC32  ATE          CPG         \n",
       "5 YJNRC33  AAR          AKH         \n",
       "6 YJNRC34  AKH          AKQ         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Crosses=readxl::read_xlsx(\"~/data/trd/Crosses.xlsx\", sheet=2)\n",
    "head(Crosses<-Crosses[,c(\"Cross ID\",\"Short name 1\",\"Short name 2\")])\n",
    "tail(Crosses)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "186af08f-5c60-4f87-bde2-0641bdffb174",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "gvcf=\"/home/jnrunge/data/trd/mapped_reads/TRD.vcf.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d840fd15-d280-4651-a1c1-fca7ddbb5704",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ChrisC1\"\n",
      "[1] \"ChrisC2\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/OHLoci-ChrisC2.sbatch\"\n",
      "[1] \"ChrisC3\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/OHLoci-ChrisC3.sbatch\"\n",
      "[1] \"ChrisC4\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/OHLoci-ChrisC4.sbatch\"\n",
      "[1] \"ChrisC5\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/OHLoci-ChrisC5.sbatch\"\n",
      "[1] \"ChrisC6\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/OHLoci-ChrisC6.sbatch\"\n",
      "[1] \"ChrisC7\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/OHLoci-ChrisC7.sbatch\"\n",
      "[1] \"ChrisC8\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/OHLoci-ChrisC8.sbatch\"\n",
      "[1] \"YJNRC1\"\n",
      "[1] \"YJNRC2\"\n",
      "[1] \"YJNRC3\"\n",
      "[1] \"YJNRC4\"\n",
      "[1] \"YJNRC5\"\n",
      "[1] \"YJNRC6\"\n",
      "[1] \"YJNRC7\"\n",
      "[1] \"YJNRC8\"\n",
      "[1] \"YJNRC9\"\n",
      "[1] \"YJNRC11\"\n",
      "[1] \"YJNRC12\"\n",
      "[1] \"YJNRC14\"\n",
      "[1] \"YJNRC15\"\n",
      "[1] \"YJNRC16\"\n",
      "[1] \"YJNRC17\"\n",
      "[1] \"YJNRC18\"\n",
      "[1] \"YJNRC19\"\n",
      "[1] \"YJNRC20\"\n",
      "[1] \"YJNRC21\"\n",
      "[1] \"YJNRC22\"\n",
      "[1] \"YJNRC23\"\n",
      "[1] \"YJNRC24\"\n",
      "[1] \"YJNRC25\"\n",
      "[1] \"YJNRC26\"\n",
      "[1] \"YJNRC27\"\n",
      "[1] \"YJNRC29\"\n",
      "[1] \"YJNRC30\"\n",
      "[1] \"YJNRC31\"\n",
      "[1] \"YJNRC32\"\n",
      "[1] \"YJNRC34\"\n",
      "[1] \"/home/jnrunge/data/trd/mapped_reads/scripts/OHLoci.list\"\n",
      "[1] \"7 /home/jnrunge/data/trd/mapped_reads/scripts/OHLoci.list\"\n",
      "[1] \"[1] \\\"Currently running 0 jobs.\\\"\"               \n",
      "[2] \"[1] \\\"7 sbatches remaining! Running 11 more...\\\"\"\n",
      "[3] \"[1] \\\"Submitted batch job 502710\\\"\"              \n",
      "[4] \"[1] \\\"Submitted batch job 502711\\\"\"              \n",
      "[5] \"[1] \\\"Submitted batch job 502712\\\"\"              \n",
      "[6] \"[1] \\\"Submitted batch job 502714\\\"\"              \n",
      "[7] \"[1] \\\"Submitted batch job 502716\\\"\"              \n"
     ]
    }
   ],
   "source": [
    "crosses=unlist(lapply(basename(bam_files), getFirst_v2, split=\".\"))\n",
    "initial_timedate=Sys.time()\n",
    "jobname=\"OHLoci\"\n",
    "scripts_dir=\"/home/jnrunge/data/trd/mapped_reads/scripts/\"\n",
    "for(cross in crosses){\n",
    "    run=FALSE\n",
    "    print(cross)\n",
    "    #gvcf<-paste0(\"/home/jnrunge/data/trd/mapped_reads/\",cross,\".g.vcf.gz\")\n",
    "    if(file.exists(ohfile<-paste0(\"~/data/trd/mapped_reads/\",cross,\".hetLoci.gz\"))){\n",
    "        if(file.mtime(gvcf)>file.mtime(ohfile)){\n",
    "            run=TRUE\n",
    "            \n",
    "        }\n",
    "    }else{\n",
    "        run=TRUE\n",
    "    }\n",
    "    if(run == TRUE){\n",
    "            file.create(running_file<-paste0(\"~/data/TRD/Oppo-Homo-Pos/\",cross,\".runningOHLOCI\"))\n",
    "            cmd=paste0(\"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci2.sh \",cross)\n",
    "            cmd=paste0(cmd,\" && rm -f \",running_file)\n",
    "            sbatch_list=execute_complex_sbatch(cmd, jobname = jobname, scripts_dir = scripts_dir, uniqueRunID = cross, cores=\"1\", mem=\"8G\", time=\"short\", env=\"bwaetc\", initial_timedate = initial_timedate, jobs_simul = 10, jobs_total = 30)\n",
    "    }\n",
    "    }\n",
    "if(exists(\"sbatch_list\") &  jobname == \"OHLoci\"){\n",
    "    print(sbatch_list)\n",
    "    start_sbatch_list(sbatch_list, 10, jobname, initial_timedate)\n",
    "}\n",
    "while(length(list.files(path = \"~/data/TRD/Oppo-Homo-Pos\", pattern = \".runningOHLOCI$\"))>0){\n",
    "    Sys.sleep(60)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "247ce53d-af7a-4d09-a291-c228d71c5740",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"BAK\"\n",
      "[1] \"BAN\"\n",
      "[1] \"AKH\"\n",
      "[1] \"ABS\"\n",
      "[1] \"AMM\"\n",
      "[1] \"ABE\"\n",
      "[1] \"BFC\"\n",
      "[1] \"BID\"\n",
      "[1] \"ACT\"\n",
      "[1] \"AKV\"\n",
      "[1] \"AMD\"\n",
      "[1] \"ANI\"\n",
      "[1] \"AND\"\n",
      "[1] \"ANH\"\n",
      "[1] \"AVI\"\n",
      "[1] \"ACI\"\n",
      "[1] \"ABC\"\n",
      "[1] \"ACN\"\n",
      "[1] \"AAM\"\n",
      "[1] \"ABG\"\n",
      "[1] \"AAR\"\n",
      "[1] \"ABA\"\n",
      "[1] \"BEF\"\n",
      "[1] \"AKI\"\n",
      "[1] \"ATE\"\n",
      "[1] \"BET\"\n",
      "[1] \"BTI\"\n",
      "[1] \"BQC\"\n",
      "[1] \"BMK\"\n",
      "[1] \"CRB\"\n",
      "[1] \"BKL\"\n",
      "[1] \"ABL\"\n",
      "[1] \"CCF\"\n",
      "[1] \"CCC\"\n",
      "[1] \"CCG\"\n",
      "[1] \"CMQ\"\n",
      "[1] \"BFQ\"\n",
      "[1] \"BFR\"\n",
      "[1] \"BHN\"\n",
      "[1] \"BFD\"\n",
      "[1] \"ABP\"\n",
      "[1] \"AKS\"\n",
      "[1] \"AKQ\"\n",
      "[1] \"CPG\"\n",
      "[1] \"ACP\"\n",
      "[1] \"BAP\"\n",
      "[1] \"CCD\"\n",
      "[1] \"ACK\"\n",
      "[1] \"AKE\"\n",
      "[1] \"BAH\"\n",
      "[1] \"ANG\"\n",
      "[1] \"BFP\"\n",
      "[1] \"CMP\"\n",
      "[1] \"SACE_YCR\"\n",
      "[1] \"ACV\"\n",
      "[1] \"CGD\"\n",
      "[1] \"CEI\"\n"
     ]
    }
   ],
   "source": [
    "jobname=\"OHLoci2\"\n",
    "remove(\"sbatch_list\")\n",
    "for(s in samples$id){\n",
    "    if(grepl(\"YJNRC\",s,fixed = TRUE)|grepl(\"Chris\",s,fixed = TRUE)){\n",
    "        next\n",
    "    }\n",
    "    run=FALSE\n",
    "    print(s)\n",
    "    if(file.exists(ohfile<-paste0(\"~/data/trd/mapped_reads/\",s,\".homLoci.gz\"))){\n",
    "        if(file.mtime(gvcf)>file.mtime(ohfile)){\n",
    "            run=TRUE\n",
    "            \n",
    "        }\n",
    "    }else{\n",
    "        run=TRUE\n",
    "    }\n",
    "    if(run == TRUE){\n",
    "            file.create(running_file<-paste0(\"~/data/TRD/Oppo-Homo-Pos/\",s,\".runningOHLOCI\"))\n",
    "            cmd=paste0(\"cd /home/jnrunge/data/trd/mapped_reads/ && sh ~/TRD/01_Mapping/02_OHLoci3.sh \",s)\n",
    "            cmd=paste0(cmd,\" && rm -f \",running_file)\n",
    "            sbatch_list=execute_complex_sbatch(cmd, jobname = jobname, scripts_dir = scripts_dir, uniqueRunID = s, cores=\"1\", mem=\"8G\", time=\"short\", env=\"bwaetc\", initial_timedate = initial_timedate, jobs_simul = 10, jobs_total = 20)\n",
    "        Sys.sleep(1)\n",
    "        }\n",
    "    }\n",
    "if(exists(\"sbatch_list\") &  jobname == \"OHLoci2\"){\n",
    "    print(sbatch_list)\n",
    "    start_sbatch_list(sbatch_list, 5, jobname, initial_timedate)\n",
    "}\n",
    "while(length(list.files(path = \"~/data/TRD/Oppo-Homo-Pos\", pattern = \".runningOHLOCI$\"))>0){\n",
    "    Sys.sleep(60)\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
