{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bfff5537",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "source(\"scripts/functions.R\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ece457ab",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "\"multiple methods tables found for 'union'\"\n",
      "Warning message:\n",
      "\"multiple methods tables found for 'intersect'\"\n",
      "\n",
      "Attaching package: 'InterMineR'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    intersect, union\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(InterMineR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "725db87b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "An object of class \"Service\"\n",
       "Slot \"mine\":\n",
       "                                    YeastMine \n",
       "\"https://yeastmine.yeastgenome.org/yeastmine\" \n",
       "\n",
       "Slot \"token\":\n",
       "[1] \"\"\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "im <- initInterMine(mine=listMines()[\"YeastMine\"])\n",
    "im"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4f95ca4d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>name</th><th scope=col>title</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td><span style=white-space:pre-wrap>Genes_Proteins_MolWt   </span></td><td>Retrieve--&gt;Proteins in a molecular weight range</td></tr>\n",
       "\t<tr><th scope=row>2</th><td><span style=white-space:pre-wrap>Gene_YeastParalogs     </span></td><td><span style=white-space:pre-wrap>Organism --&gt; All S. cerevisiae Paralogs        </span></td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Gene_ChromosomeLocation</td><td><span style=white-space:pre-wrap>Gene --&gt; Chromosomal location                  </span></td></tr>\n",
       "\t<tr><th scope=row>4</th><td><span style=white-space:pre-wrap>Pathway_Genes          </span></td><td><span style=white-space:pre-wrap>Pathway--&gt;Genes                                </span></td></tr>\n",
       "\t<tr><th scope=row>5</th><td><span style=white-space:pre-wrap>Gene_Homologs          </span></td><td>Gene --&gt; Non-Fungal and S. cerevisiae Homologs </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>DOTerm_DOTermIdentifier</td><td><span style=white-space:pre-wrap>DO Term name --&gt; DO Term Identifier            </span></td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & name & title\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Genes\\_Proteins\\_MolWt    & Retrieve-->Proteins in a molecular weight range\\\\\n",
       "\t2 & Gene\\_YeastParalogs      & Organism --> All S. cerevisiae Paralogs        \\\\\n",
       "\t3 & Gene\\_ChromosomeLocation & Gene --> Chromosomal location                  \\\\\n",
       "\t4 & Pathway\\_Genes           & Pathway-->Genes                                \\\\\n",
       "\t5 & Gene\\_Homologs           & Gene --> Non-Fungal and S. cerevisiae Homologs \\\\\n",
       "\t6 & DOTerm\\_DOTermIdentifier & DO Term name --> DO Term Identifier            \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | name &lt;chr&gt; | title &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | Genes_Proteins_MolWt    | Retrieve--&gt;Proteins in a molecular weight range |\n",
       "| 2 | Gene_YeastParalogs      | Organism --&gt; All S. cerevisiae Paralogs         |\n",
       "| 3 | Gene_ChromosomeLocation | Gene --&gt; Chromosomal location                   |\n",
       "| 4 | Pathway_Genes           | Pathway--&gt;Genes                                 |\n",
       "| 5 | Gene_Homologs           | Gene --&gt; Non-Fungal and S. cerevisiae Homologs  |\n",
       "| 6 | DOTerm_DOTermIdentifier | DO Term name --&gt; DO Term Identifier             |\n",
       "\n"
      ],
      "text/plain": [
       "  name                    title                                          \n",
       "1 Genes_Proteins_MolWt    Retrieve-->Proteins in a molecular weight range\n",
       "2 Gene_YeastParalogs      Organism --> All S. cerevisiae Paralogs        \n",
       "3 Gene_ChromosomeLocation Gene --> Chromosomal location                  \n",
       "4 Pathway_Genes           Pathway-->Genes                                \n",
       "5 Gene_Homologs           Gene --> Non-Fungal and S. cerevisiae Homologs \n",
       "6 DOTerm_DOTermIdentifier DO Term name --> DO Term Identifier            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "template = getTemplates(im)\n",
    "head(template)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "912d2d34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>name</th><th scope=col>title</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>50</th><td>ChromosomeRegion_AllGenes</td><td>Chromosomal Region --&gt; All genes</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & name & title\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t50 & ChromosomeRegion\\_AllGenes & Chromosomal Region --> All genes\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 2\n",
       "\n",
       "| <!--/--> | name &lt;chr&gt; | title &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 50 | ChromosomeRegion_AllGenes | Chromosomal Region --&gt; All genes |\n",
       "\n"
      ],
      "text/plain": [
       "   name                      title                           \n",
       "50 ChromosomeRegion_AllGenes Chromosomal Region --> All genes"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "template[startsWith(template$title, \"Chromosomal\"),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "59f9e202",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$model</dt>\n",
       "\t\t<dd><strong>name:</strong> 'genomic'</dd>\n",
       "\t<dt>$title</dt>\n",
       "\t\t<dd>'Chromosomal Region --&gt; All genes'</dd>\n",
       "\t<dt>$description</dt>\n",
       "\t\t<dd>'Retrieve all &lt;a href = \"https://sites.google.com/view/yeastgenome-help/analyze-help/yeastmine\"&gt;genes&lt;/a&gt; located within the specified chromosome and coordinate range.'</dd>\n",
       "\t<dt>$select</dt>\n",
       "\t\t<dd><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Gene.chromosome.primaryIdentifier'</li><li>'Gene.primaryIdentifier'</li><li>'Gene.secondaryIdentifier'</li><li>'Gene.featureType'</li><li>'Gene.symbol'</li><li>'Gene.name'</li><li>'Gene.sgdAlias'</li><li>'Gene.organism.shortName'</li><li>'Gene.qualifier'</li><li>'Gene.chromosomeLocation.start'</li><li>'Gene.chromosomeLocation.end'</li><li>'Gene.chromosomeLocation.strand'</li><li>'Gene.description'</li></ol>\n",
       "</dd>\n",
       "\t<dt>$constraintLogic</dt>\n",
       "\t\t<dd>'(A or B) and C and D and E and I'</dd>\n",
       "\t<dt>$name</dt>\n",
       "\t\t<dd>'ChromosomeRegion_AllGenes'</dd>\n",
       "\t<dt>$comment</dt>\n",
       "\t\t<dd>''</dd>\n",
       "\t<dt>$orderBy</dt>\n",
       "\t\t<dd><ol>\n",
       "\t<li><strong>Gene.primaryIdentifier:</strong> 'ASC'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$where</dt>\n",
       "\t\t<dd><ol>\n",
       "\t<li><dl>\n",
       "\t<dt>$path</dt>\n",
       "\t\t<dd>'Gene.status'</dd>\n",
       "\t<dt>$op</dt>\n",
       "\t\t<dd>'='</dd>\n",
       "\t<dt>$code</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>$editable</dt>\n",
       "\t\t<dd>FALSE</dd>\n",
       "\t<dt>$switchable</dt>\n",
       "\t\t<dd>FALSE</dd>\n",
       "\t<dt>$switched</dt>\n",
       "\t\t<dd>'LOCKED'</dd>\n",
       "\t<dt>$value</dt>\n",
       "\t\t<dd>'Active'</dd>\n",
       "</dl>\n",
       "</li>\n",
       "\t<li><dl>\n",
       "\t<dt>$path</dt>\n",
       "\t\t<dd>'Gene.status'</dd>\n",
       "\t<dt>$op</dt>\n",
       "\t\t<dd>'IS NULL'</dd>\n",
       "\t<dt>$code</dt>\n",
       "\t\t<dd>'B'</dd>\n",
       "\t<dt>$editable</dt>\n",
       "\t\t<dd>FALSE</dd>\n",
       "\t<dt>$switchable</dt>\n",
       "\t\t<dd>FALSE</dd>\n",
       "\t<dt>$switched</dt>\n",
       "\t\t<dd>'LOCKED'</dd>\n",
       "\t<dt>$value</dt>\n",
       "\t\t<dd>'IS NULL'</dd>\n",
       "</dl>\n",
       "</li>\n",
       "\t<li><dl>\n",
       "\t<dt>$path</dt>\n",
       "\t\t<dd>'Gene.featureType'</dd>\n",
       "\t<dt>$op</dt>\n",
       "\t\t<dd>'!='</dd>\n",
       "\t<dt>$code</dt>\n",
       "\t\t<dd>'I'</dd>\n",
       "\t<dt>$editable</dt>\n",
       "\t\t<dd>FALSE</dd>\n",
       "\t<dt>$switchable</dt>\n",
       "\t\t<dd>FALSE</dd>\n",
       "\t<dt>$switched</dt>\n",
       "\t\t<dd>'LOCKED'</dd>\n",
       "\t<dt>$value</dt>\n",
       "\t\t<dd>'not in systematic sequence of S288C'</dd>\n",
       "</dl>\n",
       "</li>\n",
       "\t<li><dl>\n",
       "\t<dt>$path</dt>\n",
       "\t\t<dd>'Gene.chromosome.primaryIdentifier'</dd>\n",
       "\t<dt>$op</dt>\n",
       "\t\t<dd>'='</dd>\n",
       "\t<dt>$code</dt>\n",
       "\t\t<dd>'C'</dd>\n",
       "\t<dt>$editable</dt>\n",
       "\t\t<dd>TRUE</dd>\n",
       "\t<dt>$switchable</dt>\n",
       "\t\t<dd>FALSE</dd>\n",
       "\t<dt>$switched</dt>\n",
       "\t\t<dd>'LOCKED'</dd>\n",
       "\t<dt>$value</dt>\n",
       "\t\t<dd>'chrXIV'</dd>\n",
       "</dl>\n",
       "</li>\n",
       "\t<li><dl>\n",
       "\t<dt>$path</dt>\n",
       "\t\t<dd>'Gene.chromosomeLocation.start'</dd>\n",
       "\t<dt>$op</dt>\n",
       "\t\t<dd>'&gt;='</dd>\n",
       "\t<dt>$code</dt>\n",
       "\t\t<dd>'D'</dd>\n",
       "\t<dt>$editable</dt>\n",
       "\t\t<dd>TRUE</dd>\n",
       "\t<dt>$switchable</dt>\n",
       "\t\t<dd>FALSE</dd>\n",
       "\t<dt>$switched</dt>\n",
       "\t\t<dd>'LOCKED'</dd>\n",
       "\t<dt>$value</dt>\n",
       "\t\t<dd>'1'</dd>\n",
       "</dl>\n",
       "</li>\n",
       "\t<li><dl>\n",
       "\t<dt>$path</dt>\n",
       "\t\t<dd>'Gene.chromosomeLocation.end'</dd>\n",
       "\t<dt>$op</dt>\n",
       "\t\t<dd>'&lt;='</dd>\n",
       "\t<dt>$code</dt>\n",
       "\t\t<dd>'E'</dd>\n",
       "\t<dt>$editable</dt>\n",
       "\t\t<dd>TRUE</dd>\n",
       "\t<dt>$switchable</dt>\n",
       "\t\t<dd>FALSE</dd>\n",
       "\t<dt>$switched</dt>\n",
       "\t\t<dd>'LOCKED'</dd>\n",
       "\t<dt>$value</dt>\n",
       "\t\t<dd>'20000'</dd>\n",
       "</dl>\n",
       "</li>\n",
       "</ol>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$model] \\textbf{name:} 'genomic'\n",
       "\\item[\\$title] 'Chromosomal Region --> All genes'\n",
       "\\item[\\$description] 'Retrieve all <a href = \"https://sites.google.com/view/yeastgenome-help/analyze-help/yeastmine\">genes</a> located within the specified chromosome and coordinate range.'\n",
       "\\item[\\$select] \\begin{enumerate*}\n",
       "\\item 'Gene.chromosome.primaryIdentifier'\n",
       "\\item 'Gene.primaryIdentifier'\n",
       "\\item 'Gene.secondaryIdentifier'\n",
       "\\item 'Gene.featureType'\n",
       "\\item 'Gene.symbol'\n",
       "\\item 'Gene.name'\n",
       "\\item 'Gene.sgdAlias'\n",
       "\\item 'Gene.organism.shortName'\n",
       "\\item 'Gene.qualifier'\n",
       "\\item 'Gene.chromosomeLocation.start'\n",
       "\\item 'Gene.chromosomeLocation.end'\n",
       "\\item 'Gene.chromosomeLocation.strand'\n",
       "\\item 'Gene.description'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$constraintLogic] '(A or B) and C and D and E and I'\n",
       "\\item[\\$name] 'ChromosomeRegion\\_AllGenes'\n",
       "\\item[\\$comment] ''\n",
       "\\item[\\$orderBy] \\begin{enumerate}\n",
       "\\item \\textbf{Gene.primaryIdentifier:} 'ASC'\n",
       "\\end{enumerate}\n",
       "\n",
       "\\item[\\$where] \\begin{enumerate}\n",
       "\\item \\begin{description}\n",
       "\\item[\\$path] 'Gene.status'\n",
       "\\item[\\$op] '='\n",
       "\\item[\\$code] 'A'\n",
       "\\item[\\$editable] FALSE\n",
       "\\item[\\$switchable] FALSE\n",
       "\\item[\\$switched] 'LOCKED'\n",
       "\\item[\\$value] 'Active'\n",
       "\\end{description}\n",
       "\n",
       "\\item \\begin{description}\n",
       "\\item[\\$path] 'Gene.status'\n",
       "\\item[\\$op] 'IS NULL'\n",
       "\\item[\\$code] 'B'\n",
       "\\item[\\$editable] FALSE\n",
       "\\item[\\$switchable] FALSE\n",
       "\\item[\\$switched] 'LOCKED'\n",
       "\\item[\\$value] 'IS NULL'\n",
       "\\end{description}\n",
       "\n",
       "\\item \\begin{description}\n",
       "\\item[\\$path] 'Gene.featureType'\n",
       "\\item[\\$op] '!='\n",
       "\\item[\\$code] 'I'\n",
       "\\item[\\$editable] FALSE\n",
       "\\item[\\$switchable] FALSE\n",
       "\\item[\\$switched] 'LOCKED'\n",
       "\\item[\\$value] 'not in systematic sequence of S288C'\n",
       "\\end{description}\n",
       "\n",
       "\\item \\begin{description}\n",
       "\\item[\\$path] 'Gene.chromosome.primaryIdentifier'\n",
       "\\item[\\$op] '='\n",
       "\\item[\\$code] 'C'\n",
       "\\item[\\$editable] TRUE\n",
       "\\item[\\$switchable] FALSE\n",
       "\\item[\\$switched] 'LOCKED'\n",
       "\\item[\\$value] 'chrXIV'\n",
       "\\end{description}\n",
       "\n",
       "\\item \\begin{description}\n",
       "\\item[\\$path] 'Gene.chromosomeLocation.start'\n",
       "\\item[\\$op] '>='\n",
       "\\item[\\$code] 'D'\n",
       "\\item[\\$editable] TRUE\n",
       "\\item[\\$switchable] FALSE\n",
       "\\item[\\$switched] 'LOCKED'\n",
       "\\item[\\$value] '1'\n",
       "\\end{description}\n",
       "\n",
       "\\item \\begin{description}\n",
       "\\item[\\$path] 'Gene.chromosomeLocation.end'\n",
       "\\item[\\$op] '<='\n",
       "\\item[\\$code] 'E'\n",
       "\\item[\\$editable] TRUE\n",
       "\\item[\\$switchable] FALSE\n",
       "\\item[\\$switched] 'LOCKED'\n",
       "\\item[\\$value] '20000'\n",
       "\\end{description}\n",
       "\n",
       "\\end{enumerate}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$model\n",
       ":   **name:** 'genomic'\n",
       "$title\n",
       ":   'Chromosomal Region --&gt; All genes'\n",
       "$description\n",
       ":   'Retrieve all &lt;a href = \"https://sites.google.com/view/yeastgenome-help/analyze-help/yeastmine\"&gt;genes&lt;/a&gt; located within the specified chromosome and coordinate range.'\n",
       "$select\n",
       ":   1. 'Gene.chromosome.primaryIdentifier'\n",
       "2. 'Gene.primaryIdentifier'\n",
       "3. 'Gene.secondaryIdentifier'\n",
       "4. 'Gene.featureType'\n",
       "5. 'Gene.symbol'\n",
       "6. 'Gene.name'\n",
       "7. 'Gene.sgdAlias'\n",
       "8. 'Gene.organism.shortName'\n",
       "9. 'Gene.qualifier'\n",
       "10. 'Gene.chromosomeLocation.start'\n",
       "11. 'Gene.chromosomeLocation.end'\n",
       "12. 'Gene.chromosomeLocation.strand'\n",
       "13. 'Gene.description'\n",
       "\n",
       "\n",
       "\n",
       "$constraintLogic\n",
       ":   '(A or B) and C and D and E and I'\n",
       "$name\n",
       ":   'ChromosomeRegion_AllGenes'\n",
       "$comment\n",
       ":   ''\n",
       "$orderBy\n",
       ":   1. **Gene.primaryIdentifier:** 'ASC'\n",
       "\n",
       "\n",
       "\n",
       "$where\n",
       ":   1. $path\n",
       ":   'Gene.status'\n",
       "$op\n",
       ":   '='\n",
       "$code\n",
       ":   'A'\n",
       "$editable\n",
       ":   FALSE\n",
       "$switchable\n",
       ":   FALSE\n",
       "$switched\n",
       ":   'LOCKED'\n",
       "$value\n",
       ":   'Active'\n",
       "\n",
       "\n",
       "\n",
       "2. $path\n",
       ":   'Gene.status'\n",
       "$op\n",
       ":   'IS NULL'\n",
       "$code\n",
       ":   'B'\n",
       "$editable\n",
       ":   FALSE\n",
       "$switchable\n",
       ":   FALSE\n",
       "$switched\n",
       ":   'LOCKED'\n",
       "$value\n",
       ":   'IS NULL'\n",
       "\n",
       "\n",
       "\n",
       "3. $path\n",
       ":   'Gene.featureType'\n",
       "$op\n",
       ":   '!='\n",
       "$code\n",
       ":   'I'\n",
       "$editable\n",
       ":   FALSE\n",
       "$switchable\n",
       ":   FALSE\n",
       "$switched\n",
       ":   'LOCKED'\n",
       "$value\n",
       ":   'not in systematic sequence of S288C'\n",
       "\n",
       "\n",
       "\n",
       "4. $path\n",
       ":   'Gene.chromosome.primaryIdentifier'\n",
       "$op\n",
       ":   '='\n",
       "$code\n",
       ":   'C'\n",
       "$editable\n",
       ":   TRUE\n",
       "$switchable\n",
       ":   FALSE\n",
       "$switched\n",
       ":   'LOCKED'\n",
       "$value\n",
       ":   'chrXIV'\n",
       "\n",
       "\n",
       "\n",
       "5. $path\n",
       ":   'Gene.chromosomeLocation.start'\n",
       "$op\n",
       ":   '&gt;='\n",
       "$code\n",
       ":   'D'\n",
       "$editable\n",
       ":   TRUE\n",
       "$switchable\n",
       ":   FALSE\n",
       "$switched\n",
       ":   'LOCKED'\n",
       "$value\n",
       ":   '1'\n",
       "\n",
       "\n",
       "\n",
       "6. $path\n",
       ":   'Gene.chromosomeLocation.end'\n",
       "$op\n",
       ":   '&lt;='\n",
       "$code\n",
       ":   'E'\n",
       "$editable\n",
       ":   TRUE\n",
       "$switchable\n",
       ":   FALSE\n",
       "$switched\n",
       ":   'LOCKED'\n",
       "$value\n",
       ":   '20000'\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$model\n",
       "     name \n",
       "\"genomic\" \n",
       "\n",
       "$title\n",
       "[1] \"Chromosomal Region --> All genes\"\n",
       "\n",
       "$description\n",
       "[1] \"Retrieve all <a href = \\\"https://sites.google.com/view/yeastgenome-help/analyze-help/yeastmine\\\">genes</a> located within the specified chromosome and coordinate range.\"\n",
       "\n",
       "$select\n",
       " [1] \"Gene.chromosome.primaryIdentifier\" \"Gene.primaryIdentifier\"           \n",
       " [3] \"Gene.secondaryIdentifier\"          \"Gene.featureType\"                 \n",
       " [5] \"Gene.symbol\"                       \"Gene.name\"                        \n",
       " [7] \"Gene.sgdAlias\"                     \"Gene.organism.shortName\"          \n",
       " [9] \"Gene.qualifier\"                    \"Gene.chromosomeLocation.start\"    \n",
       "[11] \"Gene.chromosomeLocation.end\"       \"Gene.chromosomeLocation.strand\"   \n",
       "[13] \"Gene.description\"                 \n",
       "\n",
       "$constraintLogic\n",
       "[1] \"(A or B) and C and D and E and I\"\n",
       "\n",
       "$name\n",
       "[1] \"ChromosomeRegion_AllGenes\"\n",
       "\n",
       "$comment\n",
       "[1] \"\"\n",
       "\n",
       "$orderBy\n",
       "$orderBy[[1]]\n",
       "Gene.primaryIdentifier \n",
       "                 \"ASC\" \n",
       "\n",
       "\n",
       "$where\n",
       "$where[[1]]\n",
       "$where[[1]]$path\n",
       "[1] \"Gene.status\"\n",
       "\n",
       "$where[[1]]$op\n",
       "[1] \"=\"\n",
       "\n",
       "$where[[1]]$code\n",
       "[1] \"A\"\n",
       "\n",
       "$where[[1]]$editable\n",
       "[1] FALSE\n",
       "\n",
       "$where[[1]]$switchable\n",
       "[1] FALSE\n",
       "\n",
       "$where[[1]]$switched\n",
       "[1] \"LOCKED\"\n",
       "\n",
       "$where[[1]]$value\n",
       "[1] \"Active\"\n",
       "\n",
       "\n",
       "$where[[2]]\n",
       "$where[[2]]$path\n",
       "[1] \"Gene.status\"\n",
       "\n",
       "$where[[2]]$op\n",
       "[1] \"IS NULL\"\n",
       "\n",
       "$where[[2]]$code\n",
       "[1] \"B\"\n",
       "\n",
       "$where[[2]]$editable\n",
       "[1] FALSE\n",
       "\n",
       "$where[[2]]$switchable\n",
       "[1] FALSE\n",
       "\n",
       "$where[[2]]$switched\n",
       "[1] \"LOCKED\"\n",
       "\n",
       "$where[[2]]$value\n",
       "[1] \"IS NULL\"\n",
       "\n",
       "\n",
       "$where[[3]]\n",
       "$where[[3]]$path\n",
       "[1] \"Gene.featureType\"\n",
       "\n",
       "$where[[3]]$op\n",
       "[1] \"!=\"\n",
       "\n",
       "$where[[3]]$code\n",
       "[1] \"I\"\n",
       "\n",
       "$where[[3]]$editable\n",
       "[1] FALSE\n",
       "\n",
       "$where[[3]]$switchable\n",
       "[1] FALSE\n",
       "\n",
       "$where[[3]]$switched\n",
       "[1] \"LOCKED\"\n",
       "\n",
       "$where[[3]]$value\n",
       "[1] \"not in systematic sequence of S288C\"\n",
       "\n",
       "\n",
       "$where[[4]]\n",
       "$where[[4]]$path\n",
       "[1] \"Gene.chromosome.primaryIdentifier\"\n",
       "\n",
       "$where[[4]]$op\n",
       "[1] \"=\"\n",
       "\n",
       "$where[[4]]$code\n",
       "[1] \"C\"\n",
       "\n",
       "$where[[4]]$editable\n",
       "[1] TRUE\n",
       "\n",
       "$where[[4]]$switchable\n",
       "[1] FALSE\n",
       "\n",
       "$where[[4]]$switched\n",
       "[1] \"LOCKED\"\n",
       "\n",
       "$where[[4]]$value\n",
       "[1] \"chrXIV\"\n",
       "\n",
       "\n",
       "$where[[5]]\n",
       "$where[[5]]$path\n",
       "[1] \"Gene.chromosomeLocation.start\"\n",
       "\n",
       "$where[[5]]$op\n",
       "[1] \">=\"\n",
       "\n",
       "$where[[5]]$code\n",
       "[1] \"D\"\n",
       "\n",
       "$where[[5]]$editable\n",
       "[1] TRUE\n",
       "\n",
       "$where[[5]]$switchable\n",
       "[1] FALSE\n",
       "\n",
       "$where[[5]]$switched\n",
       "[1] \"LOCKED\"\n",
       "\n",
       "$where[[5]]$value\n",
       "[1] \"1\"\n",
       "\n",
       "\n",
       "$where[[6]]\n",
       "$where[[6]]$path\n",
       "[1] \"Gene.chromosomeLocation.end\"\n",
       "\n",
       "$where[[6]]$op\n",
       "[1] \"<=\"\n",
       "\n",
       "$where[[6]]$code\n",
       "[1] \"E\"\n",
       "\n",
       "$where[[6]]$editable\n",
       "[1] TRUE\n",
       "\n",
       "$where[[6]]$switchable\n",
       "[1] FALSE\n",
       "\n",
       "$where[[6]]$switched\n",
       "[1] \"LOCKED\"\n",
       "\n",
       "$where[[6]]$value\n",
       "[1] \"20000\"\n",
       "\n",
       "\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "queryRegion = getTemplateQuery(\n",
    "  im = im, \n",
    "  name = \"ChromosomeRegion_AllGenes\"\n",
    ")\n",
    "queryRegion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0f3b7bc2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Gene.chromosome.primaryIdentifier</th><th scope=col>Gene.primaryIdentifier</th><th scope=col>Gene.secondaryIdentifier</th><th scope=col>Gene.featureType</th><th scope=col>Gene.symbol</th><th scope=col>Gene.name</th><th scope=col>Gene.sgdAlias</th><th scope=col>Gene.organism.shortName</th><th scope=col>Gene.qualifier</th><th scope=col>Gene.chromosomeLocation.start</th><th scope=col>Gene.chromosomeLocation.end</th><th scope=col>Gene.chromosomeLocation.strand</th><th scope=col>Gene.description</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chrXIV</td><td>S000005274</td><td>YNL330C</td><td>ORF</td><td>RPD3 </td><td>Reduced Potassium Dependency   </td><td>histone deacetylase RPD3|MOF6|REC3|SDI2|SDS6                 </td><td>S. cerevisiae</td><td>Verified</td><td>18001</td><td>19302</td><td>-1</td><td>Histone deacetylase, component of both the Rpd3S and Rpd3L complexes; regulates transcription, silencing, autophagy and other processes by influencing chromatin remodeling; forms at least two different complexes which have distinct functions and members; Rpd3(L) recruitment to the subtelomeric region is regulated by interaction with the arginine methyltransferase, Hmt1p</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chrXIV</td><td>S000005275</td><td>YNL331C</td><td>ORF</td><td>AAD14</td><td>Aryl-Alcohol Dehydrogenase     </td><td>putative aryl-alcohol dehydrogenase                          </td><td>S. cerevisiae</td><td>Verified</td><td>16118</td><td>17248</td><td>-1</td><td>Putative aryl-alcohol dehydrogenase; similar to P. chrysosporium aryl-alcohol dehydrogenase; mutational analysis has not yet revealed a physiological role; members of the AAD gene family comprise three pairs (AAD3 + AAD15, AAD6/AAD16 + AAD4, AAD10 + AAD14) whose two genes are more related to one another than to other members of the family                                </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chrXIV</td><td>S000005276</td><td>YNL332W</td><td>ORF</td><td>THI12</td><td>THIamine metabolism            </td><td>4-amino-5-hydroxymethyl-2-methylpyrimidine phosphate synthase</td><td>S. cerevisiae</td><td>Verified</td><td>14832</td><td>15854</td><td>1 </td><td>Protein involved in synthesis of the thiamine precursor HMP; member of a subtelomeric gene family including THI5, THI11, THI12, and THI13; hydroxymethylpyrimidine is also known as HMP                                                                                                                                                                                             </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chrXIV</td><td>S000005277</td><td>YNL333W</td><td>ORF</td><td>SNZ2 </td><td>SNooZe                         </td><td>pyridoxine biosynthesis protein SNZ2                         </td><td>S. cerevisiae</td><td>Verified</td><td>13267</td><td>14163</td><td>1 </td><td>Protein involved in thiamine and pyridoxine biosynthesis; member of a stationary phase-induced gene family where transcriptional induction precedes the diauxic shift; induced in the absence of thiamin in a Thi2/3p-dependent manner and repressed in its presence; forms a co-regulated gene pair with SNO2; interacts with Thi11p; paralog of SNZ1 and SNZ3                     </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chrXIV</td><td>S000005278</td><td>YNL334C</td><td>ORF</td><td>SNO2 </td><td>SNZ proximal Open reading frame</td><td>putative pyridoxal 5'-phosphate synthase                     </td><td>S. cerevisiae</td><td>Verified</td><td>12208</td><td>12876</td><td>-1</td><td>Protein of unknown function; nearly identical to Sno3p; expression is induced before the diauxic shift and also in the absence of thiamin                                                                                                                                                                                                                                           </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chrXIV</td><td>S000005279</td><td>YNL335W</td><td>ORF</td><td>DDI2 </td><td>DNA Damage Inducible           </td><td>cyanamide hydratase|DDI3                                     </td><td>S. cerevisiae</td><td>Verified</td><td>11452</td><td>12129</td><td>1 </td><td>Cyanamide hydratase that detoxifies cyanamide; member of the HD domain metalloprotein superfamily; expression is induced over 100-fold by cyanamide and by SN2-type DNA alkylating agents such as MMS and DMA; induction decreased in rad6 and rad18 mutants; gene and protein are identical to DDI3 and Ddi3p                                                                      </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & Gene.chromosome.primaryIdentifier & Gene.primaryIdentifier & Gene.secondaryIdentifier & Gene.featureType & Gene.symbol & Gene.name & Gene.sgdAlias & Gene.organism.shortName & Gene.qualifier & Gene.chromosomeLocation.start & Gene.chromosomeLocation.end & Gene.chromosomeLocation.strand & Gene.description\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chrXIV & S000005274 & YNL330C & ORF & RPD3  & Reduced Potassium Dependency    & histone deacetylase RPD3\\textbar{}MOF6\\textbar{}REC3\\textbar{}SDI2\\textbar{}SDS6                  & S. cerevisiae & Verified & 18001 & 19302 & -1 & Histone deacetylase, component of both the Rpd3S and Rpd3L complexes; regulates transcription, silencing, autophagy and other processes by influencing chromatin remodeling; forms at least two different complexes which have distinct functions and members; Rpd3(L) recruitment to the subtelomeric region is regulated by interaction with the arginine methyltransferase, Hmt1p\\\\\n",
       "\t2 & chrXIV & S000005275 & YNL331C & ORF & AAD14 & Aryl-Alcohol Dehydrogenase      & putative aryl-alcohol dehydrogenase                           & S. cerevisiae & Verified & 16118 & 17248 & -1 & Putative aryl-alcohol dehydrogenase; similar to P. chrysosporium aryl-alcohol dehydrogenase; mutational analysis has not yet revealed a physiological role; members of the AAD gene family comprise three pairs (AAD3 + AAD15, AAD6/AAD16 + AAD4, AAD10 + AAD14) whose two genes are more related to one another than to other members of the family                                \\\\\n",
       "\t3 & chrXIV & S000005276 & YNL332W & ORF & THI12 & THIamine metabolism             & 4-amino-5-hydroxymethyl-2-methylpyrimidine phosphate synthase & S. cerevisiae & Verified & 14832 & 15854 & 1  & Protein involved in synthesis of the thiamine precursor HMP; member of a subtelomeric gene family including THI5, THI11, THI12, and THI13; hydroxymethylpyrimidine is also known as HMP                                                                                                                                                                                             \\\\\n",
       "\t4 & chrXIV & S000005277 & YNL333W & ORF & SNZ2  & SNooZe                          & pyridoxine biosynthesis protein SNZ2                          & S. cerevisiae & Verified & 13267 & 14163 & 1  & Protein involved in thiamine and pyridoxine biosynthesis; member of a stationary phase-induced gene family where transcriptional induction precedes the diauxic shift; induced in the absence of thiamin in a Thi2/3p-dependent manner and repressed in its presence; forms a co-regulated gene pair with SNO2; interacts with Thi11p; paralog of SNZ1 and SNZ3                     \\\\\n",
       "\t5 & chrXIV & S000005278 & YNL334C & ORF & SNO2  & SNZ proximal Open reading frame & putative pyridoxal 5'-phosphate synthase                      & S. cerevisiae & Verified & 12208 & 12876 & -1 & Protein of unknown function; nearly identical to Sno3p; expression is induced before the diauxic shift and also in the absence of thiamin                                                                                                                                                                                                                                           \\\\\n",
       "\t6 & chrXIV & S000005279 & YNL335W & ORF & DDI2  & DNA Damage Inducible            & cyanamide hydratase\\textbar{}DDI3                                      & S. cerevisiae & Verified & 11452 & 12129 & 1  & Cyanamide hydratase that detoxifies cyanamide; member of the HD domain metalloprotein superfamily; expression is induced over 100-fold by cyanamide and by SN2-type DNA alkylating agents such as MMS and DMA; induction decreased in rad6 and rad18 mutants; gene and protein are identical to DDI3 and Ddi3p                                                                      \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 13\n",
       "\n",
       "| <!--/--> | Gene.chromosome.primaryIdentifier &lt;chr&gt; | Gene.primaryIdentifier &lt;chr&gt; | Gene.secondaryIdentifier &lt;chr&gt; | Gene.featureType &lt;chr&gt; | Gene.symbol &lt;chr&gt; | Gene.name &lt;chr&gt; | Gene.sgdAlias &lt;chr&gt; | Gene.organism.shortName &lt;chr&gt; | Gene.qualifier &lt;chr&gt; | Gene.chromosomeLocation.start &lt;chr&gt; | Gene.chromosomeLocation.end &lt;chr&gt; | Gene.chromosomeLocation.strand &lt;chr&gt; | Gene.description &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chrXIV | S000005274 | YNL330C | ORF | RPD3  | Reduced Potassium Dependency    | histone deacetylase RPD3|MOF6|REC3|SDI2|SDS6                  | S. cerevisiae | Verified | 18001 | 19302 | -1 | Histone deacetylase, component of both the Rpd3S and Rpd3L complexes; regulates transcription, silencing, autophagy and other processes by influencing chromatin remodeling; forms at least two different complexes which have distinct functions and members; Rpd3(L) recruitment to the subtelomeric region is regulated by interaction with the arginine methyltransferase, Hmt1p |\n",
       "| 2 | chrXIV | S000005275 | YNL331C | ORF | AAD14 | Aryl-Alcohol Dehydrogenase      | putative aryl-alcohol dehydrogenase                           | S. cerevisiae | Verified | 16118 | 17248 | -1 | Putative aryl-alcohol dehydrogenase; similar to P. chrysosporium aryl-alcohol dehydrogenase; mutational analysis has not yet revealed a physiological role; members of the AAD gene family comprise three pairs (AAD3 + AAD15, AAD6/AAD16 + AAD4, AAD10 + AAD14) whose two genes are more related to one another than to other members of the family                                 |\n",
       "| 3 | chrXIV | S000005276 | YNL332W | ORF | THI12 | THIamine metabolism             | 4-amino-5-hydroxymethyl-2-methylpyrimidine phosphate synthase | S. cerevisiae | Verified | 14832 | 15854 | 1  | Protein involved in synthesis of the thiamine precursor HMP; member of a subtelomeric gene family including THI5, THI11, THI12, and THI13; hydroxymethylpyrimidine is also known as HMP                                                                                                                                                                                              |\n",
       "| 4 | chrXIV | S000005277 | YNL333W | ORF | SNZ2  | SNooZe                          | pyridoxine biosynthesis protein SNZ2                          | S. cerevisiae | Verified | 13267 | 14163 | 1  | Protein involved in thiamine and pyridoxine biosynthesis; member of a stationary phase-induced gene family where transcriptional induction precedes the diauxic shift; induced in the absence of thiamin in a Thi2/3p-dependent manner and repressed in its presence; forms a co-regulated gene pair with SNO2; interacts with Thi11p; paralog of SNZ1 and SNZ3                      |\n",
       "| 5 | chrXIV | S000005278 | YNL334C | ORF | SNO2  | SNZ proximal Open reading frame | putative pyridoxal 5'-phosphate synthase                      | S. cerevisiae | Verified | 12208 | 12876 | -1 | Protein of unknown function; nearly identical to Sno3p; expression is induced before the diauxic shift and also in the absence of thiamin                                                                                                                                                                                                                                            |\n",
       "| 6 | chrXIV | S000005279 | YNL335W | ORF | DDI2  | DNA Damage Inducible            | cyanamide hydratase|DDI3                                      | S. cerevisiae | Verified | 11452 | 12129 | 1  | Cyanamide hydratase that detoxifies cyanamide; member of the HD domain metalloprotein superfamily; expression is induced over 100-fold by cyanamide and by SN2-type DNA alkylating agents such as MMS and DMA; induction decreased in rad6 and rad18 mutants; gene and protein are identical to DDI3 and Ddi3p                                                                       |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene.chromosome.primaryIdentifier Gene.primaryIdentifier\n",
       "1 chrXIV                            S000005274            \n",
       "2 chrXIV                            S000005275            \n",
       "3 chrXIV                            S000005276            \n",
       "4 chrXIV                            S000005277            \n",
       "5 chrXIV                            S000005278            \n",
       "6 chrXIV                            S000005279            \n",
       "  Gene.secondaryIdentifier Gene.featureType Gene.symbol\n",
       "1 YNL330C                  ORF              RPD3       \n",
       "2 YNL331C                  ORF              AAD14      \n",
       "3 YNL332W                  ORF              THI12      \n",
       "4 YNL333W                  ORF              SNZ2       \n",
       "5 YNL334C                  ORF              SNO2       \n",
       "6 YNL335W                  ORF              DDI2       \n",
       "  Gene.name                      \n",
       "1 Reduced Potassium Dependency   \n",
       "2 Aryl-Alcohol Dehydrogenase     \n",
       "3 THIamine metabolism            \n",
       "4 SNooZe                         \n",
       "5 SNZ proximal Open reading frame\n",
       "6 DNA Damage Inducible           \n",
       "  Gene.sgdAlias                                                \n",
       "1 histone deacetylase RPD3|MOF6|REC3|SDI2|SDS6                 \n",
       "2 putative aryl-alcohol dehydrogenase                          \n",
       "3 4-amino-5-hydroxymethyl-2-methylpyrimidine phosphate synthase\n",
       "4 pyridoxine biosynthesis protein SNZ2                         \n",
       "5 putative pyridoxal 5'-phosphate synthase                     \n",
       "6 cyanamide hydratase|DDI3                                     \n",
       "  Gene.organism.shortName Gene.qualifier Gene.chromosomeLocation.start\n",
       "1 S. cerevisiae           Verified       18001                        \n",
       "2 S. cerevisiae           Verified       16118                        \n",
       "3 S. cerevisiae           Verified       14832                        \n",
       "4 S. cerevisiae           Verified       13267                        \n",
       "5 S. cerevisiae           Verified       12208                        \n",
       "6 S. cerevisiae           Verified       11452                        \n",
       "  Gene.chromosomeLocation.end Gene.chromosomeLocation.strand\n",
       "1 19302                       -1                            \n",
       "2 17248                       -1                            \n",
       "3 15854                       1                             \n",
       "4 14163                       1                             \n",
       "5 12876                       -1                            \n",
       "6 12129                       1                             \n",
       "  Gene.description                                                                                                                                                                                                                                                                                                                                                                    \n",
       "1 Histone deacetylase, component of both the Rpd3S and Rpd3L complexes; regulates transcription, silencing, autophagy and other processes by influencing chromatin remodeling; forms at least two different complexes which have distinct functions and members; Rpd3(L) recruitment to the subtelomeric region is regulated by interaction with the arginine methyltransferase, Hmt1p\n",
       "2 Putative aryl-alcohol dehydrogenase; similar to P. chrysosporium aryl-alcohol dehydrogenase; mutational analysis has not yet revealed a physiological role; members of the AAD gene family comprise three pairs (AAD3 + AAD15, AAD6/AAD16 + AAD4, AAD10 + AAD14) whose two genes are more related to one another than to other members of the family                                \n",
       "3 Protein involved in synthesis of the thiamine precursor HMP; member of a subtelomeric gene family including THI5, THI11, THI12, and THI13; hydroxymethylpyrimidine is also known as HMP                                                                                                                                                                                             \n",
       "4 Protein involved in thiamine and pyridoxine biosynthesis; member of a stationary phase-induced gene family where transcriptional induction precedes the diauxic shift; induced in the absence of thiamin in a Thi2/3p-dependent manner and repressed in its presence; forms a co-regulated gene pair with SNO2; interacts with Thi11p; paralog of SNZ1 and SNZ3                     \n",
       "5 Protein of unknown function; nearly identical to Sno3p; expression is induced before the diauxic shift and also in the absence of thiamin                                                                                                                                                                                                                                           \n",
       "6 Cyanamide hydratase that detoxifies cyanamide; member of the HD domain metalloprotein superfamily; expression is induced over 100-fold by cyanamide and by SN2-type DNA alkylating agents such as MMS and DMA; induction decreased in rad6 and rad18 mutants; gene and protein are identical to DDI3 and Ddi3p                                                                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "resqueryRegion <- runQuery(im, queryRegion)\n",
    "head(resqueryRegion)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9d8622a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
