{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "237cb64a-9884-48cd-ab83-39ef10f6481c",
   "metadata": {},
   "source": [
    "I want to get a list of strains that share the allele1 or allele2 per locus per cross so that I can dynamically explore the phylogeny in this regard\n",
    "\n",
    "i.e. which strains share haplotypes across the genome, of course particularly at TRD loci"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d5965251-1def-48fa-bd98-69b14a982a2f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'tidytable'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    dt, filter, lag\n",
      "\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    %in%\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'data.table'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:tidytable':\n",
      "\n",
      "    between, first, fread, last\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:tidytable':\n",
      "\n",
      "    across, add_count, add_tally, anti_join, arrange, between,\n",
      "    bind_cols, bind_rows, c_across, case_match, case_when, coalesce,\n",
      "    consecutive_id, count, cross_join, cume_dist, cur_column, cur_data,\n",
      "    cur_group_id, cur_group_rows, dense_rank, desc, distinct, filter,\n",
      "    first, full_join, group_by, group_cols, group_split, group_vars,\n",
      "    if_all, if_any, if_else, inner_join, is_grouped_df, lag, last,\n",
      "    lead, left_join, min_rank, mutate, n, n_distinct, na_if, nest_by,\n",
      "    nest_join, nth, percent_rank, pick, pull, recode, relocate, rename,\n",
      "    rename_with, right_join, row_number, rowwise, select, semi_join,\n",
      "    slice, slice_head, slice_max, slice_min, slice_sample, slice_tail,\n",
      "    summarise, summarize, tally, top_n, transmute, tribble, ungroup\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidytable)\n",
    "source(\"~/BrusselSprouts/scripts/functions.R\")\n",
    "\n",
    "files=list.files(path = \"/home/jnrunge/data/TRD/results/shiny\", pattern=\"AF.csv.gz$\", full.names = TRUE)\n",
    "\n",
    "initial_timedate=Sys.time()\n",
    "jobname=\"getLoci\"\n",
    "scripts_dir=\"/home/jnrunge/data/trd/mapped_reads/scripts/\"\n",
    "\n",
    "for(f in files){\n",
    "    df=fread(f,data.table=TRUE)\n",
    "    fwrite(select(df,chr,pos),paste0(f,\".tmp.loci\"), sep=\"\\t\", col.names = FALSE)\n",
    "    # make a subset vcf at those loci \n",
    "    if(file.exists(paste0(f,\".loci.full.vcf.gz\"))){\n",
    "        if(file.mtime(paste0(f,\".loci.full.vcf.gz\"))>file.mtime(\"/home/jnrunge/data/trd/mapped_reads/TRD.vcf.gz\") & \n",
    "          file.mtime(paste0(f,\".loci.full.vcf.gz\")) > file.mtime(f)){\n",
    "            next\n",
    "        }\n",
    "        \n",
    "    }\n",
    "    file.create(running_file<-paste0(f,\".running_getLoci\"))\n",
    "    cmd=paste0(\"bcftools view -R \",paste0(f,\".tmp.loci\"), \" -Oz -o \",paste0(f,\".loci.full.vcf.gz\"),\" ~/data/trd/full2489Matrix.vcf.gz\")\n",
    "    sbatch_list=execute_complex_sbatch(c(cmd,\n",
    "                                         paste0(\"rm -f \",running_file)), jobname = jobname, scripts_dir = scripts_dir, uniqueRunID = which(files==f), cores=\"1\", mem=\"4G\", time=\"long\", env=\"bwaetc\", initial_timedate = initial_timedate, jobs_simul = 10, jobs_total = 30)\n",
    "\n",
    "}\n",
    "\n",
    "if(exists(\"sbatch_list\") &  jobname == \"getLoci\"){\n",
    "    print(sbatch_list)\n",
    "    start_sbatch_list(sbatch_list, 10, jobname, initial_timedate)\n",
    "}\n",
    "\n",
    "while(length(list.files(path = dirname(files[1]), pattern = \".running_getLoci$\"))>0){\n",
    "    Sys.sleep(60)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "84aef61a-9d48-4939-a4ee-536fa9059a00",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "initial_timedate=Sys.time()\n",
    "\n",
    "for(f in files){\n",
    "    \n",
    "    if(file.exists(paste0(f,\".allelesharing.csv.gz\"))){\n",
    "        if(file.mtime(paste0(f,\".allelesharing.csv.gz\"))>file.mtime(\"/home/jnrunge/data/trd/mapped_reads/TRD.vcf.gz\") & \n",
    "          file.mtime(paste0(f,\".allelesharing.csv.gz\")) > file.mtime(paste0(f,\".loci.full.vcf.gz\"))){\n",
    "            next\n",
    "        }\n",
    "        \n",
    "    }\n",
    "    jobname=\"getAS\"\n",
    "    file.create(running_file<-paste0(f,\".running_getAS\"))\n",
    "    cmd=paste0(\"Rscript ~/TRD/03_GenomicSignals/01_getAlleleSharingStrains.R \", f)\n",
    "    sbatch_list=execute_complex_sbatch(c(cmd,paste0(\"rm -f \",running_file)), jobname = jobname, scripts_dir = scripts_dir, uniqueRunID = which(files==f), cores=\"1\", mem=\"8G\", time=\"long\", env=\"JupyteR4\", initial_timedate = initial_timedate, jobs_simul = 10, jobs_total = 30)\n",
    "\n",
    "}\n",
    "\n",
    "if(exists(\"sbatch_list\") &  jobname == \"getAS\"){\n",
    "    print(sbatch_list)\n",
    "    start_sbatch_list(sbatch_list, 10, jobname, initial_timedate)\n",
    "}\n",
    "\n",
    "while(length(list.files(path = dirname(files[1]), pattern = \".running_getAS$\"))>0){\n",
    "    Sys.sleep(60)\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
