{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7895806e-6095-46dd-8b01-7c59ddd7588a",
   "metadata": {},
   "source": [
    "I was not 100% sure, but most GO enrichment methods seemed to focus on DEG data, so I have essentially just taken all GO terms and their frequency over the entire genome and compare them at the TRD loci."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ee698c6d-01f5-4574-b5ef-f55bc4bd5f88",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "GFF=ape::read.gff(\"../Shiny/data/sace_R64-3-1_20210421.gff.gz\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "038542fc-980c-480d-b0be-b451049d869e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "\"package 'data.table' was built under R version 4.1.3\"\n",
      "Warning message:\n",
      "\"package 'tidytable' was built under R version 4.1.3\"\n",
      "\n",
      "Attaching package: 'tidytable'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, fread, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    dt, filter, lag\n",
      "\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    %in%\n",
      "\n",
      "\n",
      "Warning message:\n",
      "\"package 'ggplot2' was built under R version 4.1.3\"\n",
      "\n",
      "\n",
      "Warning message:\n",
      "\"package 'dplyr' was built under R version 4.1.3\"\n",
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:tidytable':\n",
      "\n",
      "    across, add_count, add_tally, anti_join, arrange, between,\n",
      "    bind_cols, bind_rows, c_across, case_match, case_when, coalesce,\n",
      "    consecutive_id, count, cross_join, cume_dist, cur_column, cur_data,\n",
      "    cur_group_id, cur_group_rows, dense_rank, desc, distinct, filter,\n",
      "    first, full_join, group_by, group_cols, group_split, group_vars,\n",
      "    if_all, if_any, if_else, inner_join, is_grouped_df, lag, last,\n",
      "    lead, left_join, min_rank, mutate, n, n_distinct, na_if, nest_by,\n",
      "    nest_join, nth, percent_rank, pick, pull, recode, relocate, rename,\n",
      "    rename_with, right_join, row_number, rowwise, select, semi_join,\n",
      "    slice, slice_head, slice_max, slice_min, slice_sample, slice_tail,\n",
      "    summarise, summarize, tally, top_n, transmute, tribble, ungroup\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Warning message:\n",
      "\"package 'naturalsort' was built under R version 4.1.3\"\n",
      "Warning message:\n",
      "\"package 'stringr' was built under R version 4.1.3\"\n"
     ]
    }
   ],
   "source": [
    "library(data.table)\n",
    "library(tidytable)\n",
    "library(ggplot2)\n",
    "    library(rrvgo)\n",
    "source(\"~/BrusselSprouts/scripts/functions.R\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "16bbcf30-43a9-4feb-ba2d-aa437b796374",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "output_dir<-\"/home/jnrunge/data/trd/GO_data\"\n",
    "if(!dir.exists(output_dir)){\n",
    "    dir.create(output_dir)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e9b6358d-ff11-4616-b08b-2086a5215d54",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tidytable: 6 x 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>lengthSNPs</th><th scope=col>chr</th><th scope=col>global_start</th><th scope=col>global_end</th><th scope=col>lengthBp</th><th scope=col>cross</th><th scope=col>start</th><th scope=col>end</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 1</td><td> 155</td><td>chromosome2 </td><td>  668344</td><td>  790905</td><td>122561</td><td>ChrisC1</td><td>444778</td><td>567339</td></tr>\n",
       "\t<tr><td> 9</td><td> 206</td><td>chromosome14</td><td> 9398197</td><td> 9591625</td><td>193428</td><td>ChrisC1</td><td>224256</td><td>417684</td></tr>\n",
       "\t<tr><td>13</td><td> 815</td><td>chromosome15</td><td>10015850</td><td>10580994</td><td>565144</td><td>ChrisC1</td><td> 66814</td><td>631958</td></tr>\n",
       "\t<tr><td>14</td><td> 235</td><td>chromosome16</td><td>11462227</td><td>11683028</td><td>220801</td><td>ChrisC1</td><td>429498</td><td>650299</td></tr>\n",
       "\t<tr><td> 4</td><td>2603</td><td>chromosome5 </td><td> 2944974</td><td> 3258307</td><td>313333</td><td>ChrisC3</td><td> 65081</td><td>378414</td></tr>\n",
       "\t<tr><td>11</td><td> 901</td><td>chromosome11</td><td> 6549018</td><td> 6720267</td><td>171249</td><td>ChrisC3</td><td> 18001</td><td>189250</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tidytable: 6 x 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " ID & lengthSNPs & chr & global\\_start & global\\_end & lengthBp & cross & start & end\\\\\n",
       " <int> & <int> & <chr> & <int> & <int> & <int> & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  1 &  155 & chromosome2  &   668344 &   790905 & 122561 & ChrisC1 & 444778 & 567339\\\\\n",
       "\t  9 &  206 & chromosome14 &  9398197 &  9591625 & 193428 & ChrisC1 & 224256 & 417684\\\\\n",
       "\t 13 &  815 & chromosome15 & 10015850 & 10580994 & 565144 & ChrisC1 &  66814 & 631958\\\\\n",
       "\t 14 &  235 & chromosome16 & 11462227 & 11683028 & 220801 & ChrisC1 & 429498 & 650299\\\\\n",
       "\t  4 & 2603 & chromosome5  &  2944974 &  3258307 & 313333 & ChrisC3 &  65081 & 378414\\\\\n",
       "\t 11 &  901 & chromosome11 &  6549018 &  6720267 & 171249 & ChrisC3 &  18001 & 189250\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tidytable: 6 x 9\n",
       "\n",
       "| ID &lt;int&gt; | lengthSNPs &lt;int&gt; | chr &lt;chr&gt; | global_start &lt;int&gt; | global_end &lt;int&gt; | lengthBp &lt;int&gt; | cross &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "|  1 |  155 | chromosome2  |   668344 |   790905 | 122561 | ChrisC1 | 444778 | 567339 |\n",
       "|  9 |  206 | chromosome14 |  9398197 |  9591625 | 193428 | ChrisC1 | 224256 | 417684 |\n",
       "| 13 |  815 | chromosome15 | 10015850 | 10580994 | 565144 | ChrisC1 |  66814 | 631958 |\n",
       "| 14 |  235 | chromosome16 | 11462227 | 11683028 | 220801 | ChrisC1 | 429498 | 650299 |\n",
       "|  4 | 2603 | chromosome5  |  2944974 |  3258307 | 313333 | ChrisC3 |  65081 | 378414 |\n",
       "| 11 |  901 | chromosome11 |  6549018 |  6720267 | 171249 | ChrisC3 |  18001 | 189250 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID lengthSNPs chr          global_start global_end lengthBp cross   start \n",
       "1  1  155       chromosome2    668344       790905   122561   ChrisC1 444778\n",
       "2  9  206       chromosome14  9398197      9591625   193428   ChrisC1 224256\n",
       "3 13  815       chromosome15 10015850     10580994   565144   ChrisC1  66814\n",
       "4 14  235       chromosome16 11462227     11683028   220801   ChrisC1 429498\n",
       "5  4 2603       chromosome5   2944974      3258307   313333   ChrisC3  65081\n",
       "6 11  901       chromosome11  6549018      6720267   171249   ChrisC3  18001\n",
       "  end   \n",
       "1 567339\n",
       "2 417684\n",
       "3 631958\n",
       "4 650299\n",
       "5 378414\n",
       "6 189250"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# load all TRD regions\n",
    "TRD_regions<-fread_and_bind_files(list.files(\"/home/jnrunge/data/TRD/results/shiny/\",pattern=\"TRD_regions.csv.gz$\",full.names = TRUE))\n",
    "TRD_regions<-mutate(TRD_regions, file=basename(file))%>%separate(file, c(\"cross\",\"discard\"), \"-\")%>%select(-discard)%>%filter(chr_start==chr_end)%>%\n",
    "rename(chr=chr_start)%>%select(-chr_end)\n",
    "# translate global positions to chr positions\n",
    "\n",
    "for(c in select(TRD_regions,cross)%>%distinct()%>%pull())\n",
    "    {\n",
    "    AF<-fread(paste0(\"/home/jnrunge/data/TRD/results/shiny/\",c,\"-AF.csv.gz\"))\n",
    "    \n",
    "    tmp_TRD_regions<-filter(TRD_regions, cross==c)%>%select(-any_of(\"start\"))%>%\n",
    "              left_join(select(AF, chr,pos,global_pos)%>%\n",
    "              rename(global_start=global_pos), by=c(\"chr\",\"global_start\"))%>%rename(start=pos)\n",
    "    \n",
    "    TRD_regions<-filter(TRD_regions, cross!=c)%>%bind_rows(tmp_TRD_regions)\n",
    "}\n",
    "\n",
    "TRD_regions<-mutate(TRD_regions, end=global_end-(global_start-start))\n",
    "\n",
    "head(TRD_regions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "422a4e26-cf58-49b5-8490-b16686c6c7ca",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 x 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>seqid</th><th scope=col>source</th><th scope=col>type</th><th scope=col>start</th><th scope=col>end</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>phase</th><th scope=col>attributes</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chromosome1</td><td>SGD</td><td>chromosome                    </td><td>  1</td><td>230218</td><td>NA</td><td>NA</td><td>NA</td><td>ID=chromosome1;dbxref=NCBI:BK006935.2;Name=chromosome1                                                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chromosome1</td><td>SGD</td><td>telomere                      </td><td>  1</td><td>   801</td><td>NA</td><td>- </td><td>NA</td><td>ID=TEL01L;Name=TEL01L;Note=Telomeric%20region%20on%20the%20left%20arm%20of%20Chromosome%20I%3B%20composed%20of%20an%20X%20element%20core%20sequence%2C%20X%20element%20combinatorial%20repeats%2C%20and%20a%20short%20terminal%20stretch%20of%20telomeric%20repeats;display=Telomeric%20region%20on%20the%20left%20arm%20of%20Chromosome%20I;dbxref=SGD:S000028862;curie=SGD:S000028862</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chromosome1</td><td>SGD</td><td>X_element                     </td><td>337</td><td>   801</td><td>NA</td><td>- </td><td>NA</td><td>Parent=TEL01L;Name=TEL01L_X_element                                                                                                                                                                                                                                                                                                                                                    </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chromosome1</td><td>SGD</td><td>X_element_combinatorial_repeat</td><td> 63</td><td>   336</td><td>NA</td><td>- </td><td>NA</td><td>Parent=TEL01L;Name=TEL01L_X_element_combinatorial_repeat                                                                                                                                                                                                                                                                                                                               </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chromosome1</td><td>SGD</td><td>telomeric_repeat              </td><td>  1</td><td>    62</td><td>NA</td><td>- </td><td>NA</td><td>Parent=TEL01L;Name=TEL01L_telomeric_repeat_1                                                                                                                                                                                                                                                                                                                                           </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chromosome1</td><td>SGD</td><td>telomere                      </td><td>  1</td><td>   801</td><td>NA</td><td>- </td><td>NA</td><td>ID=TEL01L_telomere;Name=TEL01L_telomere;Parent=TEL01L                                                                                                                                                                                                                                                                                                                                  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & seqid & source & type & start & end & score & strand & phase & attributes\\\\\n",
       "  & <fct> & <fct> & <fct> & <int> & <int> & <dbl> & <fct> & <fct> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chromosome1 & SGD & chromosome                     &   1 & 230218 & NA & NA & NA & ID=chromosome1;dbxref=NCBI:BK006935.2;Name=chromosome1                                                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\t2 & chromosome1 & SGD & telomere                       &   1 &    801 & NA & -  & NA & ID=TEL01L;Name=TEL01L;Note=Telomeric\\%20region\\%20on\\%20the\\%20left\\%20arm\\%20of\\%20Chromosome\\%20I\\%3B\\%20composed\\%20of\\%20an\\%20X\\%20element\\%20core\\%20sequence\\%2C\\%20X\\%20element\\%20combinatorial\\%20repeats\\%2C\\%20and\\%20a\\%20short\\%20terminal\\%20stretch\\%20of\\%20telomeric\\%20repeats;display=Telomeric\\%20region\\%20on\\%20the\\%20left\\%20arm\\%20of\\%20Chromosome\\%20I;dbxref=SGD:S000028862;curie=SGD:S000028862\\\\\n",
       "\t3 & chromosome1 & SGD & X\\_element                      & 337 &    801 & NA & -  & NA & Parent=TEL01L;Name=TEL01L\\_X\\_element                                                                                                                                                                                                                                                                                                                                                    \\\\\n",
       "\t4 & chromosome1 & SGD & X\\_element\\_combinatorial\\_repeat &  63 &    336 & NA & -  & NA & Parent=TEL01L;Name=TEL01L\\_X\\_element\\_combinatorial\\_repeat                                                                                                                                                                                                                                                                                                                               \\\\\n",
       "\t5 & chromosome1 & SGD & telomeric\\_repeat               &   1 &     62 & NA & -  & NA & Parent=TEL01L;Name=TEL01L\\_telomeric\\_repeat\\_1                                                                                                                                                                                                                                                                                                                                           \\\\\n",
       "\t6 & chromosome1 & SGD & telomere                       &   1 &    801 & NA & -  & NA & ID=TEL01L\\_telomere;Name=TEL01L\\_telomere;Parent=TEL01L                                                                                                                                                                                                                                                                                                                                  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 9\n",
       "\n",
       "| <!--/--> | seqid &lt;fct&gt; | source &lt;fct&gt; | type &lt;fct&gt; | start &lt;int&gt; | end &lt;int&gt; | score &lt;dbl&gt; | strand &lt;fct&gt; | phase &lt;fct&gt; | attributes &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chromosome1 | SGD | chromosome                     |   1 | 230218 | NA | NA | NA | ID=chromosome1;dbxref=NCBI:BK006935.2;Name=chromosome1                                                                                                                                                                                                                                                                                                                                  |\n",
       "| 2 | chromosome1 | SGD | telomere                       |   1 |    801 | NA | -  | NA | ID=TEL01L;Name=TEL01L;Note=Telomeric%20region%20on%20the%20left%20arm%20of%20Chromosome%20I%3B%20composed%20of%20an%20X%20element%20core%20sequence%2C%20X%20element%20combinatorial%20repeats%2C%20and%20a%20short%20terminal%20stretch%20of%20telomeric%20repeats;display=Telomeric%20region%20on%20the%20left%20arm%20of%20Chromosome%20I;dbxref=SGD:S000028862;curie=SGD:S000028862 |\n",
       "| 3 | chromosome1 | SGD | X_element                      | 337 |    801 | NA | -  | NA | Parent=TEL01L;Name=TEL01L_X_element                                                                                                                                                                                                                                                                                                                                                     |\n",
       "| 4 | chromosome1 | SGD | X_element_combinatorial_repeat |  63 |    336 | NA | -  | NA | Parent=TEL01L;Name=TEL01L_X_element_combinatorial_repeat                                                                                                                                                                                                                                                                                                                                |\n",
       "| 5 | chromosome1 | SGD | telomeric_repeat               |   1 |     62 | NA | -  | NA | Parent=TEL01L;Name=TEL01L_telomeric_repeat_1                                                                                                                                                                                                                                                                                                                                            |\n",
       "| 6 | chromosome1 | SGD | telomere                       |   1 |    801 | NA | -  | NA | ID=TEL01L_telomere;Name=TEL01L_telomere;Parent=TEL01L                                                                                                                                                                                                                                                                                                                                   |\n",
       "\n"
      ],
      "text/plain": [
       "  seqid       source type                           start end    score strand\n",
       "1 chromosome1 SGD    chromosome                       1   230218 NA    NA    \n",
       "2 chromosome1 SGD    telomere                         1      801 NA    -     \n",
       "3 chromosome1 SGD    X_element                      337      801 NA    -     \n",
       "4 chromosome1 SGD    X_element_combinatorial_repeat  63      336 NA    -     \n",
       "5 chromosome1 SGD    telomeric_repeat                 1       62 NA    -     \n",
       "6 chromosome1 SGD    telomere                         1      801 NA    -     \n",
       "  phase\n",
       "1 NA   \n",
       "2 NA   \n",
       "3 NA   \n",
       "4 NA   \n",
       "5 NA   \n",
       "6 NA   \n",
       "  attributes                                                                                                                                                                                                                                                                                                                                                                             \n",
       "1 ID=chromosome1;dbxref=NCBI:BK006935.2;Name=chromosome1                                                                                                                                                                                                                                                                                                                                 \n",
       "2 ID=TEL01L;Name=TEL01L;Note=Telomeric%20region%20on%20the%20left%20arm%20of%20Chromosome%20I%3B%20composed%20of%20an%20X%20element%20core%20sequence%2C%20X%20element%20combinatorial%20repeats%2C%20and%20a%20short%20terminal%20stretch%20of%20telomeric%20repeats;display=Telomeric%20region%20on%20the%20left%20arm%20of%20Chromosome%20I;dbxref=SGD:S000028862;curie=SGD:S000028862\n",
       "3 Parent=TEL01L;Name=TEL01L_X_element                                                                                                                                                                                                                                                                                                                                                    \n",
       "4 Parent=TEL01L;Name=TEL01L_X_element_combinatorial_repeat                                                                                                                                                                                                                                                                                                                               \n",
       "5 Parent=TEL01L;Name=TEL01L_telomeric_repeat_1                                                                                                                                                                                                                                                                                                                                           \n",
       "6 ID=TEL01L_telomere;Name=TEL01L_telomere;Parent=TEL01L                                                                                                                                                                                                                                                                                                                                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(GFF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1ff314ae-dea1-42a3-97a2-1aec8c471685",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "GFF<-filter(GFF, type==\"gene\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "18fd8769-c2e7-425c-a00c-4ddf5e9c66f1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 x 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GO</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GO:0003674</td><td>2519</td></tr>\n",
       "\t<tr><td>GO:0005739</td><td>2471</td></tr>\n",
       "\t<tr><td>GO:0005737</td><td>1916</td></tr>\n",
       "\t<tr><td>GO:0005634</td><td>1882</td></tr>\n",
       "\t<tr><td>GO:0008150</td><td>1746</td></tr>\n",
       "\t<tr><td>GO:0005575</td><td>1273</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 x 2\n",
       "\\begin{tabular}{ll}\n",
       " GO & Freq\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t GO:0003674 & 2519\\\\\n",
       "\t GO:0005739 & 2471\\\\\n",
       "\t GO:0005737 & 1916\\\\\n",
       "\t GO:0005634 & 1882\\\\\n",
       "\t GO:0008150 & 1746\\\\\n",
       "\t GO:0005575 & 1273\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 x 2\n",
       "\n",
       "| GO &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|\n",
       "| GO:0003674 | 2519 |\n",
       "| GO:0005739 | 2471 |\n",
       "| GO:0005737 | 1916 |\n",
       "| GO:0005634 | 1882 |\n",
       "| GO:0008150 | 1746 |\n",
       "| GO:0005575 | 1273 |\n",
       "\n"
      ],
      "text/plain": [
       "  GO         Freq\n",
       "1 GO:0003674 2519\n",
       "2 GO:0005739 2471\n",
       "3 GO:0005737 1916\n",
       "4 GO:0005634 1882\n",
       "5 GO:0008150 1746\n",
       "6 GO:0005575 1273"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "extractGO<-function(x){\n",
    "    if(sum(grepl(\"Ontology_term\", x, fixed=TRUE))==0){\n",
    "        return(NULL)\n",
    "    }\n",
    "    x<-strsplit(x, \";\")[[1]]\n",
    "    GO<-x[which(startsWith(x, \"Ontology_term\"))]\n",
    "    GO<-str_replace(GO, fixed(\"Ontology_term=\"), \"\")\n",
    "    GO<-strsplit(GO, \",\")[[1]]\n",
    "    \n",
    "    return(GO)\n",
    "}\n",
    "\n",
    "background_GO<-unlist(lapply(pull(GFF, attributes), extractGO))\n",
    "background_GO<-as.data.frame(table(background_GO))%>%arrange(-Freq)\n",
    "colnames(background_GO)[1]<-\"GO\"\n",
    "background_GO<-as.data.table(background_GO)\n",
    "background_GO<-filter(background_GO, substr(GO, 1,2)==\"GO\")\n",
    "head(background_GO)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41a338dd-3d1d-47be-bedc-66b070313e5e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 2 terms that were not found in orgdb for BP\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 5 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 2 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 1 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 5 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 5 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 5 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 1 terms that were not found in orgdb for BP\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 6 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"No terms were found in orgdb for BP\n",
      "Check that the organism and ontology match the ones provided by orgdb\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 5 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 4 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 2 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 2 terms that were not found in orgdb for BP\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 1 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 8 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 1 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 6 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"No terms were found in orgdb for BP\n",
      "Check that the organism and ontology match the ones provided by orgdb\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 1 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 4 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 7 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 5 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 2 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 6 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 6 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 5 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"No terms were found in orgdb for BP\n",
      "Check that the organism and ontology match the ones provided by orgdb\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 6 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 5 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 6 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 1 terms that were not found in orgdb for BP\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 1 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 4 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"No terms were found in orgdb for BP\n",
      "Check that the organism and ontology match the ones provided by orgdb\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 4 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 5 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 1 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n",
      "preparing gene to GO mapping data...\n",
      "\n",
      "preparing IC data...\n",
      "\n",
      "Warning message in calculateSimMatrix(pull(GO_terms_enriched, GO), orgdb = \"org.Sc.sgd.db\", :\n",
      "\"Removed 3 terms that were not found in orgdb for BP\"\n",
      "Warning message in if (is.na(simMatrix)) {:\n",
      "\"the condition has length > 1 and only the first element will be used\"\n"
     ]
    }
   ],
   "source": [
    "for(j in 1:nrow(TRD_regions)){\n",
    "    \n",
    "    TRD_regions_slice<-slice(TRD_regions, j)\n",
    "    \n",
    "    if(file.exists(paste0(output_dir,\"/\",pull(TRD_regions_slice, cross), \".\", pull(TRD_regions_slice, ID), \".reducedTerms.RDS\"))){\n",
    "        if(file.mtime(paste0(output_dir,\"/\",pull(TRD_regions_slice, cross), \".\", pull(TRD_regions_slice, ID), \".reducedTerms.RDS\"))>max(file.mtime(list.files(\"/home/jnrunge/data/TRD/results/shiny/\",pattern=\"TRD_regions.csv.gz$\",full.names = TRUE)))){\n",
    "            next\n",
    "        }\n",
    "    }\n",
    "    \n",
    "    \n",
    "\n",
    "    GFF_subset<-filter(GFF, seqid==pull(TRD_regions_slice, chr),\n",
    "                      (start<=pull(TRD_regions_slice,start) & end>=pull(TRD_regions_slice,start)) |\n",
    "                       (start<=pull(TRD_regions_slice,end) & end>=pull(TRD_regions_slice,end)) |\n",
    "                       (start<=pull(TRD_regions_slice,start) & end>=pull(TRD_regions_slice,end)) |\n",
    "                       (start>=pull(TRD_regions_slice,start) & end<=pull(TRD_regions_slice,end)))\n",
    "\n",
    "\n",
    "\n",
    "    GO_terms<-unlist(lapply(pull(GFF_subset, attributes), extractGO))\n",
    "    \n",
    "    GO_terms<-as.data.frame(table(GO_terms))%>%arrange(-Freq)\n",
    "    if(nrow(GO_terms)<1){\n",
    "        next\n",
    "    }\n",
    "    \n",
    "    colnames(GO_terms)[1]<-\"GO\"\n",
    "    colnames(GO_terms)[2]<-\"Freq_location\"\n",
    "    GO_terms<-as.data.table(GO_terms)\n",
    "    GO_terms<-filter(GO_terms, substr(GO, 1,2)==\"GO\")\n",
    "\n",
    "    not_in_GO_terms<-filter(background_GO, !(GO %in% pull(GO_terms, GO)))%>%pull(GO)\n",
    "    GO_terms<-bind_rows(GO_terms, data.table(GO=not_in_GO_terms, Freq_location=0))\n",
    "\n",
    "\n",
    "\n",
    "    GO_terms<-full_join(GO_terms, background_GO, by=\"GO\")\n",
    "\n",
    "    GO_terms<-mutate(GO_terms, Other_location=sum(Freq_location)-Freq_location, Other=sum(Freq)-Freq)\n",
    "\n",
    "    head(GO_terms)\n",
    "\n",
    "    GO_terms<-filter(GO_terms,(Freq_location/Other_location) > (Freq/Other))\n",
    "\n",
    "    nrow(GO_terms)\n",
    "\n",
    "    GO_terms<-mutate(GO_terms, pChisq=NA)\n",
    "    for(i in 1:nrow(GO_terms)){\n",
    "        col1<-slice(GO_terms, i)%>%select(Freq_location,Other_location)%>%t()\n",
    "        col2<-slice(GO_terms, i)%>%select(Freq,Other)%>%t()\n",
    "        colnames(col1)<-\"col1\"\n",
    "        colnames(col2)<-\"col2\"\n",
    "        cols<-bind_cols(col1,col2)\n",
    "        p_chi<-chisq.test(cols, simulate.p.value = TRUE, B = round((1/(0.05/nrow(GO_terms)))*2))$p.value # 2x as many replicates as would be needed to reach corrected significance if possible\n",
    "        GO_terms<-mutate(GO_terms, pChisq=ifelse(row_number()==i,\n",
    "                                                p_chi,\n",
    "                                                pChisq))\n",
    "    }\n",
    "\n",
    "    nrow(filter(GO_terms, (Freq_location/Other_location) > (Freq/Other)))\n",
    "\n",
    "    -log10(0.05/nrow(GO_terms))\n",
    "\n",
    "    TRD_regions_slice\n",
    "\n",
    "    p<-ggplot(GO_terms, aes(-log10(pChisq)))+\n",
    "    geom_histogram(binwidth=0.1)+theme_bw(20)+ggtitle(\"Chisq GO terms vs global of over-represented GO terms\")+\n",
    "    geom_vline(xintercept=-log10(0.05/nrow(GO_terms)), color=\"red\")\n",
    "\n",
    "    saveRDS(p, file=paste0(output_dir,\"/\",pull(TRD_regions_slice, cross), \".\", pull(TRD_regions_slice, ID), \".GOpdist.RDS\"))\n",
    "\n",
    "\n",
    "    saveRDS(arrange(GO_terms, pChisq), file=paste0(output_dir,\"/\",pull(TRD_regions_slice, cross), \".\", pull(TRD_regions_slice, ID), \".GOdata.RDS\"))\n",
    "\n",
    "    \n",
    "    GO_terms_enriched<-filter(GO_terms, pChisq<(0.05/nrow(GO_terms)))\n",
    "\n",
    "\n",
    "    simMatrix <- calculateSimMatrix(pull(GO_terms_enriched, GO),\n",
    "                                    orgdb=\"org.Sc.sgd.db\", # yeast Sc \n",
    "                                    ont=\"BP\", # biological processes; could also use CC or MF\n",
    "                                    method=\"Rel\")\n",
    "    \n",
    "    if(is.na(simMatrix)){\n",
    "        next\n",
    "    }\n",
    "    if(length(simMatrix)==1){\n",
    "        next\n",
    "    }\n",
    "\n",
    "    scores <- setNames(-log10(pull(GO_terms_enriched, pChisq)), pull(GO_terms_enriched, GO))\n",
    "    reducedTerms <- reduceSimMatrix(simMatrix,\n",
    "                                    scores,\n",
    "                                    threshold=0.7,\n",
    "                                    orgdb=\"org.Sc.sgd.db\")\n",
    "\n",
    "    saveRDS(reducedTerms, file=paste0(output_dir,\"/\",pull(TRD_regions_slice, cross), \".\", pull(TRD_regions_slice, ID), \".reducedTerms.RDS\"))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83d72a4d-2cff-4b6c-9b82-b0a856b02b01",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "treemapPlot(readRDS(paste0(output_dir,\"/\",pull(TRD_regions_slice, cross), \".\", pull(TRD_regions_slice, ID), \".reducedTerms.RDS\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd0462f2-6079-453a-a89e-551f05fe21e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "wordcloudPlot(readRDS(paste0(output_dir,\"/\",pull(TRD_regions_slice, cross), \".\", pull(TRD_regions_slice, ID), \".reducedTerms.RDS\")), min.freq=1, colors=\"black\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
