{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2adddd6b-b36b-47d3-8e93-0030fec23143",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "\"package 'data.table' was built under R version 4.1.3\"\n",
      "Warning message:\n",
      "\"package 'dplyr' was built under R version 4.1.3\"\n",
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Warning message:\n",
      "\"package 'ggplot2' was built under R version 4.1.3\"\n",
      "Warning message:\n",
      "\"package 'naturalsort' was built under R version 4.1.3\"\n",
      "Warning message:\n",
      "\"package 'stringr' was built under R version 4.1.3\"\n",
      "Warning message:\n",
      "\"package 'tidytable' was built under R version 4.1.3\"\n",
      "Warning: tidytable was loaded after dplyr.\n",
      "This can lead to most dplyr functions being overwritten by tidytable functions.\n",
      "\n",
      "\n",
      "Attaching package: 'tidytable'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    across, add_count, add_tally, anti_join, arrange, between,\n",
      "    bind_cols, bind_rows, c_across, case_match, case_when, coalesce,\n",
      "    consecutive_id, count, cross_join, cume_dist, cur_column, cur_data,\n",
      "    cur_group_id, cur_group_rows, dense_rank, desc, distinct, filter,\n",
      "    first, full_join, group_by, group_cols, group_split, group_vars,\n",
      "    if_all, if_any, if_else, inner_join, is_grouped_df, lag, last,\n",
      "    lead, left_join, min_rank, mutate, n, n_distinct, na_if, nest_by,\n",
      "    nest_join, nth, percent_rank, pick, pull, recode, relocate, rename,\n",
      "    rename_with, right_join, row_number, rowwise, select, semi_join,\n",
      "    slice, slice_head, slice_max, slice_min, slice_sample, slice_tail,\n",
      "    summarise, summarize, tally, top_n, transmute, tribble, ungroup\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, fread, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    dt, filter, lag\n",
      "\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    %in%\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "source(\"../../BrusselSprouts/scripts/functions.R\")\n",
    "scripts_dir=\"/home/jnrunge/data/trd/mapped_reads/scripts/\"\n",
    "initial_timedate=Sys.time()\n",
    "library(tidytable)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b222b0cc-8478-43ae-b376-897671f47f62",
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=10, repr.plot.height=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d1efa99f-702b-4d7c-bcf2-5f8bed5452c5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "selectSimilarity=0.7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a0c7b029-6d88-43da-a348-3f1ebba3f656",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df_Strains=fread(\"../Shiny/data/Victor/operationalTable_Full2543Sace_Clades.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1507cee1-ab88-4eeb-9b24-e36d8bcd61c3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ChrisC1'</li><li>'ChrisC2'</li><li>'ChrisC3'</li><li>'ChrisC4'</li><li>'ChrisC5'</li><li>'ChrisC6'</li><li>'ChrisC7'</li><li>'ChrisC8'</li><li>'YJNRC1'</li><li>'YJNRC11'</li><li>'YJNRC12'</li><li>'YJNRC14'</li><li>'YJNRC15'</li><li>'YJNRC16'</li><li>'YJNRC17'</li><li>'YJNRC18'</li><li>'YJNRC19'</li><li>'YJNRC2'</li><li>'YJNRC20'</li><li>'YJNRC21'</li><li>'YJNRC22'</li><li>'YJNRC23'</li><li>'YJNRC24'</li><li>'YJNRC25'</li><li>'YJNRC26'</li><li>'YJNRC27'</li><li>'YJNRC29'</li><li>'YJNRC3'</li><li>'YJNRC30'</li><li>'YJNRC31'</li><li>'YJNRC32'</li><li>'YJNRC34'</li><li>'YJNRC4'</li><li>'YJNRC5'</li><li>'YJNRC6'</li><li>'YJNRC7'</li><li>'YJNRC8'</li><li>'YJNRC9'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ChrisC1'\n",
       "\\item 'ChrisC2'\n",
       "\\item 'ChrisC3'\n",
       "\\item 'ChrisC4'\n",
       "\\item 'ChrisC5'\n",
       "\\item 'ChrisC6'\n",
       "\\item 'ChrisC7'\n",
       "\\item 'ChrisC8'\n",
       "\\item 'YJNRC1'\n",
       "\\item 'YJNRC11'\n",
       "\\item 'YJNRC12'\n",
       "\\item 'YJNRC14'\n",
       "\\item 'YJNRC15'\n",
       "\\item 'YJNRC16'\n",
       "\\item 'YJNRC17'\n",
       "\\item 'YJNRC18'\n",
       "\\item 'YJNRC19'\n",
       "\\item 'YJNRC2'\n",
       "\\item 'YJNRC20'\n",
       "\\item 'YJNRC21'\n",
       "\\item 'YJNRC22'\n",
       "\\item 'YJNRC23'\n",
       "\\item 'YJNRC24'\n",
       "\\item 'YJNRC25'\n",
       "\\item 'YJNRC26'\n",
       "\\item 'YJNRC27'\n",
       "\\item 'YJNRC29'\n",
       "\\item 'YJNRC3'\n",
       "\\item 'YJNRC30'\n",
       "\\item 'YJNRC31'\n",
       "\\item 'YJNRC32'\n",
       "\\item 'YJNRC34'\n",
       "\\item 'YJNRC4'\n",
       "\\item 'YJNRC5'\n",
       "\\item 'YJNRC6'\n",
       "\\item 'YJNRC7'\n",
       "\\item 'YJNRC8'\n",
       "\\item 'YJNRC9'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ChrisC1'\n",
       "2. 'ChrisC2'\n",
       "3. 'ChrisC3'\n",
       "4. 'ChrisC4'\n",
       "5. 'ChrisC5'\n",
       "6. 'ChrisC6'\n",
       "7. 'ChrisC7'\n",
       "8. 'ChrisC8'\n",
       "9. 'YJNRC1'\n",
       "10. 'YJNRC11'\n",
       "11. 'YJNRC12'\n",
       "12. 'YJNRC14'\n",
       "13. 'YJNRC15'\n",
       "14. 'YJNRC16'\n",
       "15. 'YJNRC17'\n",
       "16. 'YJNRC18'\n",
       "17. 'YJNRC19'\n",
       "18. 'YJNRC2'\n",
       "19. 'YJNRC20'\n",
       "20. 'YJNRC21'\n",
       "21. 'YJNRC22'\n",
       "22. 'YJNRC23'\n",
       "23. 'YJNRC24'\n",
       "24. 'YJNRC25'\n",
       "25. 'YJNRC26'\n",
       "26. 'YJNRC27'\n",
       "27. 'YJNRC29'\n",
       "28. 'YJNRC3'\n",
       "29. 'YJNRC30'\n",
       "30. 'YJNRC31'\n",
       "31. 'YJNRC32'\n",
       "32. 'YJNRC34'\n",
       "33. 'YJNRC4'\n",
       "34. 'YJNRC5'\n",
       "35. 'YJNRC6'\n",
       "36. 'YJNRC7'\n",
       "37. 'YJNRC8'\n",
       "38. 'YJNRC9'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"ChrisC1\" \"ChrisC2\" \"ChrisC3\" \"ChrisC4\" \"ChrisC5\" \"ChrisC6\" \"ChrisC7\"\n",
       " [8] \"ChrisC8\" \"YJNRC1\"  \"YJNRC11\" \"YJNRC12\" \"YJNRC14\" \"YJNRC15\" \"YJNRC16\"\n",
       "[15] \"YJNRC17\" \"YJNRC18\" \"YJNRC19\" \"YJNRC2\"  \"YJNRC20\" \"YJNRC21\" \"YJNRC22\"\n",
       "[22] \"YJNRC23\" \"YJNRC24\" \"YJNRC25\" \"YJNRC26\" \"YJNRC27\" \"YJNRC29\" \"YJNRC3\" \n",
       "[29] \"YJNRC30\" \"YJNRC31\" \"YJNRC32\" \"YJNRC34\" \"YJNRC4\"  \"YJNRC5\"  \"YJNRC6\" \n",
       "[36] \"YJNRC7\"  \"YJNRC8\"  \"YJNRC9\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "crosses=readLines(\"~/data/trd/mapped_reads/TRD.vcf.gz.samples\")\n",
    "crosses=crosses[startsWith(crosses, \"YJNRC\") | startsWith(crosses, \"Chris\")]\n",
    "crosses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "39d97511-5e75-41b1-b0d8-3a9bdf3f3d22",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "file_a_exists_and_is_newer_than_file_b=function(a,b){\n",
    "    if(file.exists(a)){\n",
    "        if(file.mtime(a)>file.mtime(b)){\n",
    "            return(TRUE)\n",
    "        }\n",
    "    }\n",
    "    return(FALSE)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71e6dc8-614c-45c3-96a7-1756b0219afb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ChrisC1\"\n",
      "[1] \"ChrisC1 with TRD 1 has no or not enough or not updated pixy files :(\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "\"\u001b[1m\u001b[22m`aes_string()` was deprecated in ggplot2 3.0.0.\n",
      "\u001b[36mi\u001b[39m Please use tidy evaluation ideoms with `aes()`\"\n"
     ]
    }
   ],
   "source": [
    "for(c in crosses){\n",
    "    \n",
    "    print(c)\n",
    "\n",
    "\n",
    "    # Define file paths\n",
    "    TRD_regions_file = paste0(\"/home/jnrunge/data/TRD/results/shiny/\", c, \"-TRD_regions.csv.gz\")\n",
    "    AS_file = paste0(\"/home/jnrunge/data/TRD/results/shiny/\", c, \"-AF.csv.gz.allelesharing.csv.gz\")\n",
    "    TRD_file = paste0(\"/home/jnrunge/data/TRD/results/shiny/\", c, \"-AF.csv.gz\")\n",
    "\n",
    "    # Check if files exist and are updated\n",
    "    if (!file_a_exists_and_is_newer_than_file_b(TRD_file, \"~/data/trd/mapped_reads/TRD.vcf.gz\")) {\n",
    "        print(paste0(c, \" TRD file does not exist or is not newer than TRD.vcf.gz\"))\n",
    "        next\n",
    "    }\n",
    "\n",
    "    if (!file_a_exists_and_is_newer_than_file_b(TRD_regions_file, TRD_file)) {\n",
    "        print(paste0(c, \" has no TRD regions or no up-to-date ones\"))\n",
    "        next\n",
    "    }\n",
    "\n",
    "    if (!file_a_exists_and_is_newer_than_file_b(AS_file, TRD_file)) {\n",
    "        print(paste0(c, \" has no or no updated allele sharing file\"))\n",
    "        next\n",
    "    }\n",
    "\n",
    "    # Get all pixy files\n",
    "    pixy_files_all = list.files(path = \"~/data/trd/mapped_reads\", pattern = paste0(c, \"_.*pixy.*txt.gz$\"), full.names = TRUE)\n",
    "\n",
    "    if (length(pixy_files_all) == 0) {\n",
    "        print(paste(c, \" has no pixy files.\"))\n",
    "        next\n",
    "    }\n",
    "\n",
    "    # Load files\n",
    "    AS = fread(AS_file)\n",
    "    TRD = fread(TRD_file)\n",
    "    TRD_loci = fread(TRD_regions_file)\n",
    "\n",
    "    for(i in 1:nrow(TRD_loci)){\n",
    "        \n",
    "        id<-pull(slice(TRD_loci, i), ID)\n",
    "\n",
    "        # Select specific pixy files\n",
    "        pixy_files = list.files(path = \"~/data/trd/mapped_reads\", pattern = paste0(c, \"_\", id, \".*pixy.*txt.gz$\"), full.names = TRUE)\n",
    "\n",
    "        if (length(pixy_files) != ((16 * 3)+3)) {\n",
    "            print(paste0(c, \" with TRD \", id, \" has no or not enough or not updated pixy files :(\"))\n",
    "            next\n",
    "        }\n",
    "        \n",
    "        pixy_file_TRD_region<-pixy_files[grepl(\"theTRDregion\",pixy_files,fixed=TRUE)]\n",
    "        pixy_files<-pixy_files[pixy_files!=pixy_file_TRD_region]\n",
    "\n",
    "        # Check for chromosome overlapping TRD\n",
    "        if (TRD_loci$chr_start[i] != TRD_loci$chr_end[i]) {\n",
    "            stop(\"chr overlapping TRD\")\n",
    "        }\n",
    "\n",
    "        # Filtering and processing\n",
    "        #TRD_subset = filter(TRD, chr == TRD_loci$chr_start[i] & global_pos >= TRD_loci$global_start[i] & global_pos <= TRD_loci$global_end[i])\n",
    "        #df_AS_filtered = filter(AS, `#CHROM` == TRD_loci$chr_start[i], POS %in% TRD_subset$pos)\n",
    "\n",
    "        # Process pixy files\n",
    "        pixy_pi = fread_and_bind_files(pixy_files[grepl(\"_pi.txt\", pixy_files, fixed = TRUE)])\n",
    "        setDT(pixy_pi)\n",
    "        pixy_dxy = fread_and_bind_files(pixy_files[grepl(\"_dxy.txt\", pixy_files, fixed = TRUE)])\n",
    "        setDT(pixy_dxy)\n",
    "        pixy_fst = fread_and_bind_files(pixy_files[grepl(\"_fst.txt\", pixy_files, fixed = TRUE)])\n",
    "        setDT(pixy_fst)\n",
    "\n",
    "        pixy_pi = arrange(pixy_pi, pop)\n",
    "        pixy_pi$pop_fct = as.factor(pixy_pi$pop)\n",
    "        pixy_pi$color = NA\n",
    "        pixy_pi$color[pixy_pi$pop == \"distorter-like\"] = colorRampPalette(c(\"#FFC4CB\", \"#8B0000\"))(1)\n",
    "        pixy_pi$color[pixy_pi$pop == \"other\"] = colorRampPalette(c(\"#ADD8E6\", \"#00008B\"))(1)\n",
    "\n",
    "        # Generate plot\n",
    "        options(repr.plot.width = 10, repr.plot.height = 5)\n",
    "\n",
    "        # old way of approximating the region\n",
    "        #pi_summary<-filter(pixy_pi, chromosome == TRD_loci$chr_start[i],\n",
    "        #                              (window_pos_1>=min(TRD_subset$pos) & window_pos_2<=max(TRD_subset$pos)) | \n",
    "        #                               (window_pos_1<=min(TRD_subset$pos) & window_pos_2>=min(TRD_subset$pos)) |\n",
    "        #                               (window_pos_1<=max(TRD_subset$pos) & window_pos_2>=max(TRD_subset$pos)))%>%  group_by(pop) %>% summarise(mean_pi=mean(avg_pi,na.rm=TRUE))\n",
    "        \n",
    "        pi_summary<-fread(pixy_file_TRD_region[endsWith(pixy_file_TRD_region, \"pi.txt.gz\")], header=TRUE, sep=\"\\t\")%>%group_by(pop)%>% summarise(mean_pi=mean(avg_pi,na.rm=TRUE))\n",
    "\n",
    "        # Function definition\n",
    "        calculate_measures <- function(df, value, population) {\n",
    "\n",
    "          # Calculate pi measures\n",
    "          pi_pop <- filter(pi_summary, pop == population) %>% pull(mean_pi)\n",
    "\n",
    "          # Calculate ecdf\n",
    "          ecdf_value <- ecdf(filter(df, pop == population) %>% pull(value))(pi_pop)\n",
    "\n",
    "          # Calculate the standardized measure\n",
    "          standardized_measure <- (pi_pop - mean(filter(df, pop == population) %>% pull(value), na.rm = TRUE)) /\n",
    "                                    sd(filter(df, pop == population) %>% pull(value), na.rm = TRUE)\n",
    "\n",
    "          # Return a list of calculated values\n",
    "          return(list(\"pi_pop\" = pi_pop, \n",
    "                      \"ecdf_value\" = ecdf_value, \n",
    "                      \"standardized_measure\" = standardized_measure))\n",
    "        }\n",
    "\n",
    "        # Calculate measures for 'distorter-like' population in pixy_pi data frame using 'avg_pi' column\n",
    "        distorter_like_measures <- calculate_measures(pixy_pi, \"avg_pi\", \"distorter-like\")\n",
    "\n",
    "        # Calculate measures for 'other' population in pixy_pi data frame using 'avg_pi' column\n",
    "        other_measures <- calculate_measures(pixy_pi, \"avg_pi\", \"other\")\n",
    "\n",
    "\n",
    "        # pi plot with two colors for the two groups\n",
    "\n",
    "        p<-ggplot(pixy_pi, aes(avg_pi*100,fill=pop))+\n",
    "        geom_histogram(binwidth=0.05,alpha=1)+scale_x_log10(breaks=c(0.0001,0.001,0.01)*100, labels=c(0.0001,0.001,0.01), minor_breaks=seq(from=0,to=1*100,by=0.001*100))+\n",
    "        scale_color_manual(values=c(\"distorter-like\"=\"red\", \"other\"=\"blue\"))+\n",
    "        scale_fill_manual(values=c(\"distorter-like\"=\"red\", \"other\"=\"blue\"))+\n",
    "        geom_segment(data=pi_summary,\n",
    "                   mapping=aes(x=mean_pi*100, xend=mean_pi*100, y=-1, yend=-30, color=pop), linewidth=3)+theme_bw(18)+xlab(\"Nucleotide diversity\")+ylab(NULL)\n",
    "\n",
    "        saveRDS(p, paste0(\"/home/jnrunge/data/trd/pop_genomics/\",c,\".\",id,\"-pi-plot.RDS\"))\n",
    "\n",
    "        library(dplyr)\n",
    "        library(ggplot2)\n",
    "\n",
    "        calculate_summarize_plot <- function(df_TRDregion, df, df_name, TRD_loci, TRD_subset, i, value, xlab_text, x_scale = NULL) {\n",
    "\n",
    "          # Calculate summarize\n",
    "          sum_value <- summarise(\n",
    "            group_by(\n",
    "              df_TRDregion, pop1, pop2), mean_val=mean(get(value),na.rm=TRUE)\n",
    "          ) %>% pull(mean_val)\n",
    "\n",
    "          # Plot\n",
    "          p <- ggplot(df, aes_string(value))+\n",
    "            geom_histogram()+\n",
    "            geom_vline(xintercept = sum_value, color=\"red\")+\n",
    "            theme_bw(18)+ylab(NULL)+xlab(xlab_text)\n",
    "\n",
    "          # If x_scale is provided, apply it to the plot\n",
    "          if(!is.null(x_scale)) {\n",
    "            p <- p + x_scale\n",
    "          }\n",
    "\n",
    "          saveRDS(p, paste0(\"/home/jnrunge/data/trd/pop_genomics/\",c,\".\",id,\"-\",xlab_text,\"-plot.RDS\"))\n",
    "\n",
    "          # Calculate ecdf\n",
    "          ecdf_value <- ecdf(pull(df, value))(sum_value)\n",
    "\n",
    "          # Calculate the standardized measure\n",
    "          standardized_measure <- (sum_value - mean(pull(df, value), na.rm = TRUE)) /\n",
    "                                    sd(pull(df, value), na.rm = TRUE)\n",
    "\n",
    "          # Return a list of calculated values\n",
    "          return(list(\"sum_value\" = sum_value, \n",
    "                      \"ecdf_value\" = ecdf_value, \n",
    "                      \"standardized_measure\" = standardized_measure))\n",
    "        }\n",
    "\n",
    "        \n",
    "        \n",
    "\n",
    "        # For pixy_fst data frame\n",
    "        fst_TRDregion<-fread(pixy_file_TRD_region[endsWith(pixy_file_TRD_region, \"fst.txt.gz\")], header=TRUE, sep=\"\\t\")\n",
    "        fst_results <- calculate_summarize_plot(fst_TRDregion, pixy_fst, \"pixy_fst\", TRD_loci, TRD_subset, i, \"avg_wc_fst\", \"Fst\")\n",
    "\n",
    "        # For pixy_dxy data frame\n",
    "        dxy_TRDregion<-fread(pixy_file_TRD_region[endsWith(pixy_file_TRD_region, \"dxy.txt.gz\")], header=TRUE, sep=\"\\t\")\n",
    "        dxy_scale <- scale_x_log10(breaks=c(0.0001,0.001,0.01)*100, labels=c(0.0001,0.001,0.01), minor_breaks=seq(from=0,to=1*100,by=0.001*100))\n",
    "        dxy_results <- calculate_summarize_plot(dxy_TRDregion, pixy_dxy, \"pixy_dxy\", TRD_loci, TRD_subset, i, \"avg_dxy\", \"Dxy\", dxy_scale)\n",
    "\n",
    "\n",
    "        results_values<-list(\n",
    "        \"pi_distorter\"=distorter_like_measures,\n",
    "        \"pi_other\"=other_measures,\n",
    "        \"fst\"=fst_results,\n",
    "        \"dxy\"=dxy_results)\n",
    "\n",
    "\n",
    "\n",
    "        saveRDS(results_values, paste0(\"/home/jnrunge/data/trd/pop_genomics/\",c,\".\",id,\"-relative-measures.RDS\"))\n",
    "        \n",
    "        }\n",
    "    }\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
