{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2adddd6b-b36b-47d3-8e93-0030fec23143",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Warning: tidytable was loaded after dplyr.\n",
      "This can lead to most dplyr functions being overwritten by tidytable functions.\n",
      "\n",
      "\n",
      "Attaching package: 'tidytable'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    across, add_count, add_tally, anti_join, arrange, between,\n",
      "    bind_cols, bind_rows, c_across, case_match, case_when, coalesce,\n",
      "    consecutive_id, count, cross_join, cume_dist, cur_column, cur_data,\n",
      "    cur_group_id, cur_group_rows, dense_rank, desc, distinct, filter,\n",
      "    first, full_join, group_by, group_cols, group_split, group_vars,\n",
      "    if_all, if_any, if_else, inner_join, is_grouped_df, lag, last,\n",
      "    lead, left_join, min_rank, mutate, n, n_distinct, na_if, nest_by,\n",
      "    nest_join, nth, percent_rank, pick, pull, recode, relocate, rename,\n",
      "    rename_with, right_join, row_number, rowwise, select, semi_join,\n",
      "    slice, slice_head, slice_max, slice_min, slice_sample, slice_tail,\n",
      "    summarise, summarize, tally, top_n, transmute, tribble, ungroup\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    between, first, fread, last\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    dt, filter, lag\n",
      "\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    %in%\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "source(\"../../BrusselSprouts/scripts/functions.R\")\n",
    "scripts_dir=\"/home/jnrunge/data/trd/mapped_reads/scripts/\"\n",
    "initial_timedate=Sys.time()\n",
    "library(tidytable)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b222b0cc-8478-43ae-b376-897671f47f62",
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=10, repr.plot.height=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d1efa99f-702b-4d7c-bcf2-5f8bed5452c5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "selectSimilarity=0.7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a0c7b029-6d88-43da-a348-3f1ebba3f656",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df_Strains=fread(\"../Shiny/data/Victor/operationalTable_Full2543Sace_Clades.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1507cee1-ab88-4eeb-9b24-e36d8bcd61c3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ChrisC1'</li><li>'ChrisC2'</li><li>'ChrisC3'</li><li>'ChrisC4'</li><li>'ChrisC5'</li><li>'ChrisC6'</li><li>'ChrisC7'</li><li>'ChrisC8'</li><li>'YJNRC1'</li><li>'YJNRC11'</li><li>'YJNRC12'</li><li>'YJNRC14'</li><li>'YJNRC15'</li><li>'YJNRC16'</li><li>'YJNRC17'</li><li>'YJNRC18'</li><li>'YJNRC19'</li><li>'YJNRC2'</li><li>'YJNRC20'</li><li>'YJNRC21'</li><li>'YJNRC22'</li><li>'YJNRC23'</li><li>'YJNRC24'</li><li>'YJNRC25'</li><li>'YJNRC26'</li><li>'YJNRC27'</li><li>'YJNRC29'</li><li>'YJNRC3'</li><li>'YJNRC30'</li><li>'YJNRC31'</li><li>'YJNRC32'</li><li>'YJNRC34'</li><li>'YJNRC4'</li><li>'YJNRC5'</li><li>'YJNRC6'</li><li>'YJNRC7'</li><li>'YJNRC8'</li><li>'YJNRC9'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ChrisC1'\n",
       "\\item 'ChrisC2'\n",
       "\\item 'ChrisC3'\n",
       "\\item 'ChrisC4'\n",
       "\\item 'ChrisC5'\n",
       "\\item 'ChrisC6'\n",
       "\\item 'ChrisC7'\n",
       "\\item 'ChrisC8'\n",
       "\\item 'YJNRC1'\n",
       "\\item 'YJNRC11'\n",
       "\\item 'YJNRC12'\n",
       "\\item 'YJNRC14'\n",
       "\\item 'YJNRC15'\n",
       "\\item 'YJNRC16'\n",
       "\\item 'YJNRC17'\n",
       "\\item 'YJNRC18'\n",
       "\\item 'YJNRC19'\n",
       "\\item 'YJNRC2'\n",
       "\\item 'YJNRC20'\n",
       "\\item 'YJNRC21'\n",
       "\\item 'YJNRC22'\n",
       "\\item 'YJNRC23'\n",
       "\\item 'YJNRC24'\n",
       "\\item 'YJNRC25'\n",
       "\\item 'YJNRC26'\n",
       "\\item 'YJNRC27'\n",
       "\\item 'YJNRC29'\n",
       "\\item 'YJNRC3'\n",
       "\\item 'YJNRC30'\n",
       "\\item 'YJNRC31'\n",
       "\\item 'YJNRC32'\n",
       "\\item 'YJNRC34'\n",
       "\\item 'YJNRC4'\n",
       "\\item 'YJNRC5'\n",
       "\\item 'YJNRC6'\n",
       "\\item 'YJNRC7'\n",
       "\\item 'YJNRC8'\n",
       "\\item 'YJNRC9'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ChrisC1'\n",
       "2. 'ChrisC2'\n",
       "3. 'ChrisC3'\n",
       "4. 'ChrisC4'\n",
       "5. 'ChrisC5'\n",
       "6. 'ChrisC6'\n",
       "7. 'ChrisC7'\n",
       "8. 'ChrisC8'\n",
       "9. 'YJNRC1'\n",
       "10. 'YJNRC11'\n",
       "11. 'YJNRC12'\n",
       "12. 'YJNRC14'\n",
       "13. 'YJNRC15'\n",
       "14. 'YJNRC16'\n",
       "15. 'YJNRC17'\n",
       "16. 'YJNRC18'\n",
       "17. 'YJNRC19'\n",
       "18. 'YJNRC2'\n",
       "19. 'YJNRC20'\n",
       "20. 'YJNRC21'\n",
       "21. 'YJNRC22'\n",
       "22. 'YJNRC23'\n",
       "23. 'YJNRC24'\n",
       "24. 'YJNRC25'\n",
       "25. 'YJNRC26'\n",
       "26. 'YJNRC27'\n",
       "27. 'YJNRC29'\n",
       "28. 'YJNRC3'\n",
       "29. 'YJNRC30'\n",
       "30. 'YJNRC31'\n",
       "31. 'YJNRC32'\n",
       "32. 'YJNRC34'\n",
       "33. 'YJNRC4'\n",
       "34. 'YJNRC5'\n",
       "35. 'YJNRC6'\n",
       "36. 'YJNRC7'\n",
       "37. 'YJNRC8'\n",
       "38. 'YJNRC9'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"ChrisC1\" \"ChrisC2\" \"ChrisC3\" \"ChrisC4\" \"ChrisC5\" \"ChrisC6\" \"ChrisC7\"\n",
       " [8] \"ChrisC8\" \"YJNRC1\"  \"YJNRC11\" \"YJNRC12\" \"YJNRC14\" \"YJNRC15\" \"YJNRC16\"\n",
       "[15] \"YJNRC17\" \"YJNRC18\" \"YJNRC19\" \"YJNRC2\"  \"YJNRC20\" \"YJNRC21\" \"YJNRC22\"\n",
       "[22] \"YJNRC23\" \"YJNRC24\" \"YJNRC25\" \"YJNRC26\" \"YJNRC27\" \"YJNRC29\" \"YJNRC3\" \n",
       "[29] \"YJNRC30\" \"YJNRC31\" \"YJNRC32\" \"YJNRC34\" \"YJNRC4\"  \"YJNRC5\"  \"YJNRC6\" \n",
       "[36] \"YJNRC7\"  \"YJNRC8\"  \"YJNRC9\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "crosses=readLines(\"~/data/trd/mapped_reads/TRD.vcf.gz.samples\")\n",
    "crosses=crosses[startsWith(crosses, \"YJNRC\") | startsWith(crosses, \"Chris\")]\n",
    "crosses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a53955f4-f5f1-42ea-b838-7f1cbb8500c6",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): \n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): \nTraceback:\n",
      "1. stop()"
     ]
    }
   ],
   "source": [
    "for(c in crosses){\n",
    "    if(!file.exists(paste0(\"/home/jnrunge/data/TRD/results/shiny/\",c,\"-TRD_regions.csv.gz\"))){\n",
    "            next\n",
    "        }\n",
    "\n",
    "    AS=fread(paste0(\"/home/jnrunge/data/TRD/results/shiny/\",c,\"-AF.csv.gz.allelesharing.csv.gz\"))\n",
    "\n",
    "    TRD=fread(paste0(\"/home/jnrunge/data/TRD/results/shiny/\",c,\"-AF.csv.gz\"))\n",
    "    head(TRD)\n",
    "\n",
    "    TRD_loci=fread(paste0(\"/home/jnrunge/data/TRD/results/shiny/\",c,\"-TRD_regions.csv.gz\"))\n",
    "\n",
    "    head(AS)\n",
    "\n",
    "    TRD_loci\n",
    "\n",
    "    for(i in 1:nrow(TRD_loci)){\n",
    "\n",
    "\n",
    "        if(TRD_loci$chr_start[i]!=TRD_loci$chr_end[i]){\n",
    "            stop(\"chr overlapping TRD\")\n",
    "        }\n",
    "        TRD_subset=filter(TRD, chr== TRD_loci$chr_start[i] & global_pos >= TRD_loci$global_start[i] & global_pos <= TRD_loci$global_end[i])\n",
    "        df_AS_filtered=filter(AS, `#CHROM` == TRD_loci$chr_start[i], POS %in% TRD_subset$pos)\n",
    "        melted=reshape2::melt(df_AS_filtered, id.vars = c(\"#CHROM\",\"POS\"))\n",
    "        melted=filter(melted, variable != \"chrpos\")\n",
    "        tmp=summarise(group_by(melted, variable), nAll=n())\n",
    "        vcf_translated_summary=left_join(tmp,summarise(group_by(melted, variable, value), n=n()), by=c(\"variable\"))%>%mutate(p=n/nAll)%>%select(variable,value,p)%>%rename(Strain=variable, Type=value)\n",
    "        A1s=vcf_translated_summary$Strain[vcf_translated_summary$Type==\"A1_hom\" & vcf_translated_summary$p>=selectSimilarity]\n",
    "        A2s=vcf_translated_summary$Strain[vcf_translated_summary$Type==\"A2_hom\" & vcf_translated_summary$p>=selectSimilarity]\n",
    "        pop_summary=bind_rows(summarise(group_by(filter(df_Strains, StandardizedName %in% A1s),\n",
    "                               Clade), n=n()) %>% arrange(-n)%>%mutate(Type=\"A1_hom\"),\n",
    "                      summarise(group_by(filter(df_Strains, StandardizedName %in% A2s),\n",
    "                                         Clade), n=n()) %>% arrange(-n)%>%mutate(Type=\"A2_hom\"))%>% arrange(-n)\n",
    "        pop_summary\n",
    "\n",
    "        if(mean(TRD_subset$AD_A1/TRD_subset$sumCount)<0.5){\n",
    "            distorter=\"A2\"\n",
    "            nondistorter=\"A1\"\n",
    "        }else{\n",
    "            distorter=\"A1\"\n",
    "            nondistorter=\"A2\"}\n",
    "\n",
    "        pop_summary=filter(pop_summary, !is.na(Clade))\n",
    "        pop_summary=mutate(pop_summary, Clade=gsub(\"[ .]\",\"_\",Clade))\n",
    "\n",
    "        pop_summary$Type2=NA\n",
    "        pop_summary$Type2[pop_summary$Type==paste0(distorter,\"_hom\")]=\"Distorter-like\"\n",
    "        pop_summary$Type2[pop_summary$Type==paste0(nondistorter,\"_hom\")]=\"Non-distorter-like\"\n",
    "\n",
    "        pixy_files=list.files(path = \"~/data/trd/mapped_reads\", pattern=paste0(c,\"_\",i,\".*pixy.*txt.gz$\"), full.names = TRUE)\n",
    "        \n",
    "        if(length(pixy_files)!=(16*3)){\n",
    "            next\n",
    "        }\n",
    "\n",
    "        pixy_pi=fread_and_bind_files(pixy_files[grepl(\"_pi.txt\",pixy_files,fixed=TRUE)])\n",
    "        setDT(pixy_pi)\n",
    "        pixy_dxy=fread_and_bind_files(pixy_files[grepl(\"_dxy.txt\",pixy_files,fixed=TRUE)])\n",
    "        setDT(pixy_dxy)\n",
    "        pixy_fst=fread_and_bind_files(pixy_files[grepl(\"_fst.txt\",pixy_files,fixed=TRUE)])\n",
    "        setDT(pixy_fst)\n",
    "\n",
    "        pixy_pi=arrange(pixy_pi, pop)\n",
    "\n",
    "\n",
    "        pop_summary=arrange(pop_summary, Clade)\n",
    "\n",
    "        pop_summary$color=NA\n",
    "        pop_summary$color[pop_summary$Type2==\"Distorter-like\"]=colorRampPalette(c(\"#FFC4CB\",\"#8B0000\"))(sum(pop_summary$Type2==\"Distorter-like\"))\n",
    "        pop_summary$color[pop_summary$Type2==\"Non-distorter-like\"]=colorRampPalette(c(\"#ADD8E6\",\"#00008B\"))(sum(pop_summary$Type2==\"Non-distorter-like\"))\n",
    "\n",
    "        pop_summary\n",
    "\n",
    "        # pop_summary is not used when distorter vs others is done \n",
    "\n",
    "        pixy_pi$pop_fct=factor(pixy_pi$pop, levels=pop_summary$Clade)\n",
    "\n",
    "        # not used when only two groups\n",
    "\n",
    "\n",
    "        p<-ggplot(filter(pixy_pi, chromosome == TRD_loci$chr_start[i]), aes(avg_pi*100,fill=pop))+\n",
    "        geom_histogram(binwidth=0.05,alpha=1)+scale_x_log10(breaks=c(0.0001,0.001,0.01)*100, labels=c(0.0001,0.001,0.01), minor_breaks=seq(from=0,to=1*100,by=0.001*100))+\n",
    "        #scale_color_manual(values=pop_summary$color)+\n",
    "        #scale_fill_manual(values=pop_summary$color)+\n",
    "        scale_color_manual(values=c(\"distorter-like\"=\"red\", \"other\"=\"blue\"))+\n",
    "        scale_fill_manual(values=c(\"distorter-like\"=\"red\", \"other\"=\"blue\"))+\n",
    "        #geom_vline(data=filter(pixy_pi, chromosome == TRD_loci$chr_start[i], pop%in%pop_summary$Clade,\n",
    "        #                              (window_pos_1>=min(TRD_subset$pos) & window_pos_2<=max(TRD_subset$pos)) | \n",
    "        #                               (window_pos_1<=min(TRD_subset$pos) & window_pos_2>=min(TRD_subset$pos)) |\n",
    "        #                               (window_pos_1<=max(TRD_subset$pos) & window_pos_2>=max(TRD_subset$pos)))%>%  group_by(pop) %>% #summarise(mean_pi=mean(avg_pi)),\n",
    "        #           mapping=aes(xintercept=mean_pi*100), size=3.5,color=\"black\")+\n",
    "        geom_segment(data=filter(pixy_pi, chromosome == TRD_loci$chr_start[i],\n",
    "                                      (window_pos_1>=min(TRD_subset$pos) & window_pos_2<=max(TRD_subset$pos)) | \n",
    "                                       (window_pos_1<=min(TRD_subset$pos) & window_pos_2>=min(TRD_subset$pos)) |\n",
    "                                       (window_pos_1<=max(TRD_subset$pos) & window_pos_2>=max(TRD_subset$pos)))%>%  group_by(pop) %>% summarise(mean_pi=mean(avg_pi)),\n",
    "                   mapping=aes(x=mean_pi*100, xend=mean_pi*100, y=-1, yend=-10, color=pop), linewidth=3)+theme_bw(18)+xlab(\"Nucleotide diversity\")+ylab(NULL)\n",
    "\n",
    "        print(p)\n",
    "\n",
    "        options(repr.plot.width=10, repr.plot.height=5)\n",
    "\n",
    "        p<-ggplot(pixy_pi, aes(avg_pi*100,fill=pop))+\n",
    "        geom_histogram(binwidth=0.05,alpha=1)+scale_x_log10(breaks=c(0.0001,0.001,0.01)*100, labels=c(0.0001,0.001,0.01), minor_breaks=seq(from=0,to=1*100,by=0.001*100))+\n",
    "        #scale_color_manual(values=pop_summary$color)+\n",
    "        #scale_fill_manual(values=pop_summary$color)+\n",
    "        scale_color_manual(values=c(\"distorter-like\"=\"red\", \"other\"=\"blue\"))+\n",
    "        scale_fill_manual(values=c(\"distorter-like\"=\"red\", \"other\"=\"blue\"))+\n",
    "        #geom_vline(data=filter(pixy_pi, chromosome == TRD_loci$chr_start[i], pop%in%pop_summary$Clade,\n",
    "        #                              (window_pos_1>=min(TRD_subset$pos) & window_pos_2<=max(TRD_subset$pos)) | \n",
    "        #                               (window_pos_1<=min(TRD_subset$pos) & window_pos_2>=min(TRD_subset$pos)) |\n",
    "        #                               (window_pos_1<=max(TRD_subset$pos) & window_pos_2>=max(TRD_subset$pos)))%>%  group_by(pop) %>% #summarise(mean_pi=mean(avg_pi)),\n",
    "        #           mapping=aes(xintercept=mean_pi*100), size=3.5,color=\"black\")+\n",
    "        geom_segment(data=filter(pixy_pi, chromosome == TRD_loci$chr_start[i],\n",
    "                                      (window_pos_1>=min(TRD_subset$pos) & window_pos_2<=max(TRD_subset$pos)) | \n",
    "                                       (window_pos_1<=min(TRD_subset$pos) & window_pos_2>=min(TRD_subset$pos)) |\n",
    "                                       (window_pos_1<=max(TRD_subset$pos) & window_pos_2>=max(TRD_subset$pos)))%>%  group_by(pop) %>% summarise(mean_pi=mean(avg_pi)),\n",
    "                   mapping=aes(x=mean_pi*100, xend=mean_pi*100, y=-1, yend=-10, color=pop), linewidth=3)+theme_bw(18)+xlab(\"Nucleotide diversity\")+ylab(NULL)\n",
    "\n",
    "        print(p)\n",
    "\n",
    "        # code below works for 1 (!) distorter-like pop\n",
    "\n",
    "        sum_fst_chr<-summarise(group_by(filter(pixy_fst, chromosome == TRD_loci$chr_start[i], \n",
    "            #                                   pop1%in%pop_summary$Clade[pop_summary$Type2==\"Distorter-like\"] | \n",
    "            #   pop2%in%pop_summary$Clade[pop_summary$Type2==\"Distorter-like\"]\n",
    "                                              ),\n",
    "                                        pop1, pop2), mean_fst_chr=mean(avg_wc_fst,na.rm=TRUE))\n",
    "\n",
    "\n",
    "        sum_fst_TRD<-summarise(group_by(filter(pixy_fst, chromosome == TRD_loci$chr_start[i], \n",
    "                                               #pop1%in%pop_summary$Clade[pop_summary$Type2==\"Distorter-like\"] | \n",
    "               #pop2%in%pop_summary$Clade[pop_summary$Type2==\"Distorter-like\"],\n",
    "                                      (window_pos_1>=min(TRD_subset$pos) & window_pos_2<=max(TRD_subset$pos)) | \n",
    "                                       (window_pos_1<=min(TRD_subset$pos) & window_pos_2>=min(TRD_subset$pos)) |\n",
    "                                       (window_pos_1<=max(TRD_subset$pos) & window_pos_2>=max(TRD_subset$pos))), pop1, pop2), mean_fst_TRD=mean(avg_wc_fst,na.rm=TRUE))\n",
    "\n",
    "        sum_fst=full_join(sum_fst_chr,sum_fst_TRD)\n",
    "\n",
    "        sum_fst_melt=reshape2::melt(sum_fst, measure.vars=c(\"mean_fst_TRD\",\"mean_fst_chr\"))\n",
    "        sum_fst_melt\n",
    "\n",
    "        p<-ggplot(pixy_fst, aes(avg_wc_fst))+\n",
    "        geom_histogram()+geom_vline(xintercept = sum_fst_melt$value[sum_fst_melt$variable==\"mean_fst_TRD\"],color=\"red\")+\n",
    "        theme_bw(18)+ylab(NULL)+xlab(\"Fst\")\n",
    "\n",
    "        print(p)\n",
    "\n",
    "        #sum_fst=left_join(sum_fst, mutate(pop_summary,pop1=Clade,color1=color)%>%select(pop1,color1))\n",
    "        #sum_fst=left_join(sum_fst, mutate(pop_summary,pop2=Clade,color2=color)%>%select(pop2,color2))\n",
    "\n",
    "\n",
    "        #sum_fst$color1[is.na(sum_fst$color1)]=\"grey\"\n",
    "        #sum_fst$color2[is.na(sum_fst$color2)]=\"grey\"\n",
    "\n",
    "        #etDF(sum_fst)\n",
    "        #sum_fst$color=sum_fst$color1\n",
    "        #sum_fst$color[sum_fst$pop1==\"22__East_Asian_Wild\"]=sum_fst$color2[sum_fst$pop1==\"22__East_Asian_Wild\"]\n",
    "        #setDT(sum_fst)\n",
    "\n",
    "        #ggplot(sum_fst, aes(mean_fst_TRD-mean_fst_chr))+\n",
    "        #geom_histogram()+geom_vline(xintercept = 0)+#+scale_fill_manual(values=sum_fst$color)+\n",
    "        #theme_bw(18)+ylab(NULL)+xlab(\"Fixation index (distorter vs other clades) in TRD region vs. rest of chr\")+theme(legend.position=\"none\")\n",
    "        }\n",
    "    stop()\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
