{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2adddd6b-b36b-47d3-8e93-0030fec23143",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "source(\"../../BrusselSprouts/scripts/functions.R\")\n",
    "scripts_dir=\"/home/jnrunge/data/trd/mapped_reads/scripts/\"\n",
    "initial_timedate=Sys.time()\n",
    "library(tidytable)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b222b0cc-8478-43ae-b376-897671f47f62",
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=10, repr.plot.height=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1efa99f-702b-4d7c-bcf2-5f8bed5452c5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "selectSimilarity=0.7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a0c7b029-6d88-43da-a348-3f1ebba3f656",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df_Strains=fread(\"../Shiny/data/Victor/operationalTable_Full2543Sace_Clades.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1507cee1-ab88-4eeb-9b24-e36d8bcd61c3",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "crosses=readLines(\"~/data/trd/mapped_reads/TRD.vcf.gz.samples\")\n",
    "crosses=crosses[startsWith(crosses, \"YJNRC\") | startsWith(crosses, \"Chris\")]\n",
    "crosses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39d97511-5e75-41b1-b0d8-3a9bdf3f3d22",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "file_a_exists_and_is_newer_than_file_b=function(a,b){\n",
    "    if(file.exists(a)){\n",
    "        if(file.mtime(a)>file.mtime(b)){\n",
    "            return(TRUE)\n",
    "        }\n",
    "    }\n",
    "    return(FALSE)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71e6dc8-614c-45c3-96a7-1756b0219afb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(c in crosses){\n",
    "    \n",
    "    print(c)\n",
    "\n",
    "\n",
    "    # Define file paths\n",
    "    TRD_regions_file = paste0(\"/home/jnrunge/data/TRD/results/shiny/\", c, \"-TRD_regions.csv.gz\")\n",
    "    AS_file = paste0(\"/home/jnrunge/data/TRD/results/shiny/\", c, \"-AF.csv.gz.allelesharing.csv.gz\")\n",
    "    TRD_file = paste0(\"/home/jnrunge/data/TRD/results/shiny/\", c, \"-AF.csv.gz\")\n",
    "\n",
    "    # Check if files exist and are updated\n",
    "    if (!file_a_exists_and_is_newer_than_file_b(TRD_file, \"~/data/trd/mapped_reads/TRD.vcf.gz\")) {\n",
    "        print(paste0(c, \" TRD file does not exist or is not newer than TRD.vcf.gz\"))\n",
    "        next\n",
    "    }\n",
    "\n",
    "    if (!file_a_exists_and_is_newer_than_file_b(TRD_regions_file, TRD_file)) {\n",
    "        print(paste0(c, \" has no TRD regions or no up-to-date ones\"))\n",
    "        next\n",
    "    }\n",
    "\n",
    "    if (!file_a_exists_and_is_newer_than_file_b(AS_file, TRD_file)) {\n",
    "        print(paste0(c, \" has no or no updated allele sharing file\"))\n",
    "        next\n",
    "    }\n",
    "\n",
    "    # Get all pixy files\n",
    "    pixy_files_all = list.files(path = \"~/data/trd/mapped_reads\", pattern = paste0(c, \"_.*pixy.*txt.gz$\"), full.names = TRUE)\n",
    "\n",
    "    if (length(pixy_files_all) == 0) {\n",
    "        print(paste(c, \" has no pixy files.\"))\n",
    "        next\n",
    "    }\n",
    "\n",
    "    # Load files\n",
    "    AS = fread(AS_file)\n",
    "    TRD = fread(TRD_file)\n",
    "    TRD_loci = fread(TRD_regions_file)\n",
    "\n",
    "    for(i in 1:nrow(TRD_loci)){\n",
    "\n",
    "        # Select specific pixy files\n",
    "        pixy_files = list.files(path = \"~/data/trd/mapped_reads\", pattern = paste0(c, \"_\", i, \".*pixy.*txt.gz$\"), full.names = TRUE)\n",
    "\n",
    "        if (length(pixy_files) != (16 * 3)) {\n",
    "            print(paste0(c, \" with TRD \", i, \" has no or not enough or not updated pixy files :(\"))\n",
    "            next\n",
    "        }\n",
    "\n",
    "        # Check for chromosome overlapping TRD\n",
    "        if (TRD_loci$chr_start[i] != TRD_loci$chr_end[i]) {\n",
    "            stop(\"chr overlapping TRD\")\n",
    "        }\n",
    "\n",
    "        # Filtering and processing\n",
    "        TRD_subset = filter(TRD, chr == TRD_loci$chr_start[i] & global_pos >= TRD_loci$global_start[i] & global_pos <= TRD_loci$global_end[i])\n",
    "        df_AS_filtered = filter(AS, `#CHROM` == TRD_loci$chr_start[i], POS %in% TRD_subset$pos)\n",
    "\n",
    "        # Process pixy files\n",
    "        pixy_pi = fread_and_bind_files(pixy_files[grepl(\"_pi.txt\", pixy_files, fixed = TRUE)])\n",
    "        setDT(pixy_pi)\n",
    "        pixy_dxy = fread_and_bind_files(pixy_files[grepl(\"_dxy.txt\", pixy_files, fixed = TRUE)])\n",
    "        setDT(pixy_dxy)\n",
    "        pixy_fst = fread_and_bind_files(pixy_files[grepl(\"_fst.txt\", pixy_files, fixed = TRUE)])\n",
    "        setDT(pixy_fst)\n",
    "\n",
    "        pixy_pi = arrange(pixy_pi, pop)\n",
    "        pixy_pi$pop_fct = as.factor(pixy_pi$pop)\n",
    "        pixy_pi$color = NA\n",
    "        pixy_pi$color[pixy_pi$pop == \"distorter-like\"] = colorRampPalette(c(\"#FFC4CB\", \"#8B0000\"))(1)\n",
    "        pixy_pi$color[pixy_pi$pop == \"other\"] = colorRampPalette(c(\"#ADD8E6\", \"#00008B\"))(1)\n",
    "\n",
    "        # Generate plot\n",
    "        options(repr.plot.width = 10, repr.plot.height = 5)\n",
    "\n",
    "        pi_summary<-filter(pixy_pi, chromosome == TRD_loci$chr_start[i],\n",
    "                                      (window_pos_1>=min(TRD_subset$pos) & window_pos_2<=max(TRD_subset$pos)) | \n",
    "                                       (window_pos_1<=min(TRD_subset$pos) & window_pos_2>=min(TRD_subset$pos)) |\n",
    "                                       (window_pos_1<=max(TRD_subset$pos) & window_pos_2>=max(TRD_subset$pos)))%>%  group_by(pop) %>% summarise(mean_pi=mean(avg_pi))\n",
    "\n",
    "        # Function definition\n",
    "        calculate_measures <- function(df, value, population) {\n",
    "\n",
    "          # Calculate pi measures\n",
    "          pi_pop <- filter(pi_summary, pop == population) %>% pull(mean_pi)\n",
    "\n",
    "          # Calculate ecdf\n",
    "          ecdf_value <- ecdf(filter(df, pop == population) %>% pull(value))(pi_pop)\n",
    "\n",
    "          # Calculate the standardized measure\n",
    "          standardized_measure <- (pi_pop - mean(filter(df, pop == population) %>% pull(value), na.rm = TRUE)) /\n",
    "                                    sd(filter(df, pop == population) %>% pull(value), na.rm = TRUE)\n",
    "\n",
    "          # Return a list of calculated values\n",
    "          return(list(\"pi_pop\" = pi_pop, \n",
    "                      \"ecdf_value\" = ecdf_value, \n",
    "                      \"standardized_measure\" = standardized_measure))\n",
    "        }\n",
    "\n",
    "        # Calculate measures for 'distorter-like' population in pixy_pi data frame using 'avg_pi' column\n",
    "        distorter_like_measures <- calculate_measures(pixy_pi, \"avg_pi\", \"distorter-like\")\n",
    "\n",
    "        # Calculate measures for 'other' population in pixy_pi data frame using 'avg_pi' column\n",
    "        other_measures <- calculate_measures(pixy_pi, \"avg_pi\", \"other\")\n",
    "\n",
    "\n",
    "        # pi plot with two colors for the two groups\n",
    "\n",
    "        p<-ggplot(pixy_pi, aes(avg_pi*100,fill=pop))+\n",
    "        geom_histogram(binwidth=0.05,alpha=1)+scale_x_log10(breaks=c(0.0001,0.001,0.01)*100, labels=c(0.0001,0.001,0.01), minor_breaks=seq(from=0,to=1*100,by=0.001*100))+\n",
    "        scale_color_manual(values=c(\"distorter-like\"=\"red\", \"other\"=\"blue\"))+\n",
    "        scale_fill_manual(values=c(\"distorter-like\"=\"red\", \"other\"=\"blue\"))+\n",
    "        geom_segment(data=pi_summary,\n",
    "                   mapping=aes(x=mean_pi*100, xend=mean_pi*100, y=-1, yend=-30, color=pop), linewidth=3)+theme_bw(18)+xlab(\"Nucleotide diversity\")+ylab(NULL)\n",
    "\n",
    "        saveRDS(p, paste0(\"/home/jnrunge/data/trd/pop_genomics/\",c,\".\",i,\"-pi-plot.RDS\"))\n",
    "\n",
    "        library(dplyr)\n",
    "        library(ggplot2)\n",
    "\n",
    "        calculate_summarize_plot <- function(df, df_name, TRD_loci, TRD_subset, i, value, xlab_text, x_scale = NULL) {\n",
    "\n",
    "          # Calculate summarize\n",
    "          sum_value <- summarise(\n",
    "            group_by(\n",
    "              filter(df, chromosome == TRD_loci$chr_start[i], \n",
    "                     (window_pos_1>=min(TRD_subset$pos) & window_pos_2<=max(TRD_subset$pos)) | \n",
    "                     (window_pos_1<=min(TRD_subset$pos) & window_pos_2>=min(TRD_subset$pos)) |\n",
    "                     (window_pos_1<=max(TRD_subset$pos) & window_pos_2>=max(TRD_subset$pos))), pop1, pop2), mean_val=mean(get(value),na.rm=TRUE)\n",
    "          ) %>% pull(mean_val)\n",
    "\n",
    "          # Plot\n",
    "          p <- ggplot(df, aes_string(value))+\n",
    "            geom_histogram()+\n",
    "            geom_vline(xintercept = sum_value, color=\"red\")+\n",
    "            theme_bw(18)+ylab(NULL)+xlab(xlab_text)\n",
    "\n",
    "          # If x_scale is provided, apply it to the plot\n",
    "          if(!is.null(x_scale)) {\n",
    "            p <- p + x_scale\n",
    "          }\n",
    "\n",
    "          saveRDS(p, paste0(\"/home/jnrunge/data/trd/pop_genomics/\",c,\".\",i,\"-\",xlab_text,\"-plot.RDS\"))\n",
    "\n",
    "          # Calculate ecdf\n",
    "          ecdf_value <- ecdf(pull(df, value))(sum_value)\n",
    "\n",
    "          # Calculate the standardized measure\n",
    "          standardized_measure <- (sum_value - mean(pull(df, value), na.rm = TRUE)) /\n",
    "                                    sd(pull(df, value), na.rm = TRUE)\n",
    "\n",
    "          # Return a list of calculated values\n",
    "          return(list(\"sum_value\" = sum_value, \n",
    "                      \"ecdf_value\" = ecdf_value, \n",
    "                      \"standardized_measure\" = standardized_measure))\n",
    "        }\n",
    "\n",
    "\n",
    "        # For pixy_fst data frame\n",
    "        fst_results <- calculate_summarize_plot(pixy_fst, \"pixy_fst\", TRD_loci, TRD_subset, i, \"avg_wc_fst\", \"Fst\")\n",
    "\n",
    "        # For pixy_dxy data frame\n",
    "        dxy_scale <- scale_x_log10(breaks=c(0.0001,0.001,0.01)*100, labels=c(0.0001,0.001,0.01), minor_breaks=seq(from=0,to=1*100,by=0.001*100))\n",
    "        dxy_results <- calculate_summarize_plot(pixy_dxy, \"pixy_dxy\", TRD_loci, TRD_subset, i, \"avg_dxy\", \"Dxy\", dxy_scale)\n",
    "\n",
    "\n",
    "        results_values<-list(\n",
    "        \"pi_distorter\"=distorter_like_measures,\n",
    "        \"pi_other\"=other_measures,\n",
    "        \"fst\"=fst_results,\n",
    "        \"dxy\"=dxy_results)\n",
    "\n",
    "\n",
    "\n",
    "        saveRDS(results_values, paste0(\"/home/jnrunge/data/trd/pop_genomics/\",c,\".\",i,\"-relative-measures.RDS\"))\n",
    "        \n",
    "        }\n",
    "    }\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
